### BUSCO plot
```bash
mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Trioza/urticae/v1/BUSCO/hemiptera_odb10/short_summary.specific.hemiptera_odb10.hemiptera_odb10.txt /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco/short_summary.specific.hemiptera_odb10.Tri_urt.txt
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Dyspera/apicales/v1/BUSCO/hemiptera_odb10/short_summary.specific.hemiptera_odb10.hemiptera_odb10.txt /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco/short_summary.specific.hemiptera_odb10.Dys_api.txt
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Dyspera/pallida/v1/BUSCO/hemiptera_odb10/short_summary.specific.hemiptera_odb10.hemiptera_odb10.txt /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco/short_summary.specific.hemiptera_odb10.Dys_pal.txt
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Diaphorina/citri/GCA_000475195.1/BUSCO/hemiptera_odb10/short_summary.specific.hemiptera_odb10.hemiptera_odb10.txt /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco/short_summary.specific.hemiptera_odb10.Dia_cit.txt
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Bactericera/cockerelli/GCA_024516035.1/BUSCO/hemiptera_odb10/short_summary.specific.hemiptera_odb10.hemiptera_odb10.txt /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco/short_summary.specific.hemiptera_odb10.Bac_coc.txt
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Pachypsylla/venusta/GCA_012654025.1/BUSCO/hemiptera_odb10/short_summary.specific.hemiptera_odb10.hemiptera_odb10.txt /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco/short_summary.specific.hemiptera_odb10.Pac_ven.txt
source package 97a7f391-0f5c-4bdb-b951-14d0d5ee4576
generate_plot.py -wd /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco
```
```R
######################################
#
# BUSCO summary figure
# @version 4.0.0
# @since BUSCO 2.0.0
# 
# Copyright (c) 2016-2021, Evgeny Zdobnov (ez@ezlab.org)
# Licensed under the MIT license. See LICENSE.md file.
#
######################################

# Load the required libraries
library(ggplot2)
library("grid")

# !!! CONFIGURE YOUR PLOT HERE !!! 
# Output
my_output <- paste("/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco/","busco_figure.png",sep="/") 
my_width <- 20
my_height <- 15
my_unit <- "cm"

# Colors
my_colors <- c("#56B4E9", "#3492C7", "#F0E442", "#F04442")
# Bar height ratio
my_bar_height <- 0.75

# Legend
my_title <- "BUSCO Assessment Results"

# Font
my_family <- "sans"
my_size_ratio <- 1

# !!! SEE YOUR DATA HERE !!! 
# Your data as generated by python, remove or add more
my_species <- c('Dyspersa apicalis', 'Dyspersa apicalis', 'Dyspersa apicalis', 'Dyspersa apicalis', 'Bactericera cockerelli', 'Bactericera cockerelli', 'Bactericera cockerelli', 'Bactericera cockerelli', 'Pachypsylla venusta', 'Pachypsylla venusta', 'Pachypsylla venusta', 'Pachypsylla venusta', 'Trioza urticae', 'Trioza urticae', 'Trioza urticae', 'Trioza urticae', 'Diaphorina citri', 'Diaphorina citri', 'Diaphorina citri', 'Diaphorina citri', 'Dyspersa pallida', 'Dyspersa pallida', 'Dyspersa pallida', 'Dyspersa pallida')
my_species <- factor(my_species)
my_species <- factor(my_species,levels(my_species)[c(length(levels(my_species)):1)]) # reorder your species here just by changing the values in the vector :
my_percentage <- c(92.2, 2.4, 2.8, 2.6, 75.8, 15.4, 2.5, 6.3, 87.7, 1.6, 6.1, 4.6, 82.2, 3.0, 5.3, 9.5, 84.2, 4.7, 7.0, 4.1, 90.8, 1.6, 3.1, 4.5)
my_values <- c(2313, 60, 71, 66, 1903, 387, 64, 156, 2201, 41, 153, 115, 2062, 75, 132, 241, 2114, 117, 175, 104, 2280, 39, 78, 113)

my_species <- factor(my_species, levels = c("Pachypsylla venusta", "Diaphorina citri", "Dyspersa pallida", "Dyspersa apicalis", "Trioza urticae", "Bactericera cockerelli"))

######################################
######################################
######################################
# Code to produce the graph
labsize = 1
if (length(levels(my_species)) > 10){
  labsize = 0.66
}
print("Plotting the figure ...")
category <- c(rep(c("S","D","F","M"),c(1)))
category <-factor(category)
category = factor(category,levels(category)[c(4,1,2,3)])
df = data.frame(my_species,my_percentage,my_values,category)

figure <- ggplot() + 
  
  geom_bar(aes(y = my_percentage, x = my_species, fill = category), position = position_stack(reverse = TRUE), data = df, stat="identity", width=my_bar_height) + 
  coord_flip() + 
  theme_gray(base_size = 8) + 
  scale_y_continuous(labels = c("0","20","40","60","80","100"), breaks = c(0,20,40,60,80,100)) + 
  scale_fill_manual(values = my_colors,labels =c(" Complete (C) and single-copy (S)  ",
                                                 " Complete (C) and duplicated (D)",
                                                 " Fragmented (F)  ",
                                                 " Missing (M)")) +   
  ggtitle(my_title) + 
  xlab("") + 
  ylab("\n%BUSCOs") + 
  
  theme(plot.title = element_text(family=my_family, hjust=0.5, colour = "black", size = rel(2.2)*my_size_ratio, face = "bold")) + 
  theme(legend.position="top",legend.title = element_blank()) + 
  theme(legend.text = element_text(family=my_family, size = rel(1.2)*my_size_ratio)) + 
  theme(panel.background = element_rect(color="#FFFFFF", fill="white")) + 
  theme(panel.grid.minor = element_blank()) + 
  theme(panel.grid.major = element_blank()) +
  theme(axis.text.y = element_text(family=my_family, face = "italic", colour = "black", size = rel(1.66)*my_size_ratio)) + 
  theme(axis.text.x = element_text(family=my_family, colour = "black", size = rel(1.66)*my_size_ratio)) + 
  theme(axis.line = element_line(size=1*my_size_ratio, colour = "black")) + 
  theme(axis.ticks.length = unit(.85, "cm")) + 
  theme(axis.ticks.y = element_line(colour="white", size = 0)) + 
  theme(axis.ticks.x = element_line(colour="#222222")) + 
  theme(axis.ticks.length = unit(0.4, "cm")) + 
  theme(axis.title.x = element_text(family=my_family, size=rel(1.2)*my_size_ratio)) + 
  
  guides(fill = guide_legend(override.aes = list(colour = NULL))) +
  guides(fill=guide_legend(nrow=2,byrow=TRUE))

for(i in rev(c(1:length(levels(my_species))))){
  detailed_values <- my_values[my_species==my_species[my_species==levels(my_species)[i]]]
  total_buscos <- sum(detailed_values)
  figure <- figure + 
    annotate("text", label=paste("C:", detailed_values[1] + detailed_values[2], " [S:", detailed_values[1], ", D:", detailed_values[2], "], F:", detailed_values[3], ", M:", detailed_values[4], ", n:", total_buscos, sep=""), 
             y=3, x = i, size = labsize*4*my_size_ratio, colour = "black", hjust=0, family=my_family)
}

figure
```
### Phylogeny
```bash
mkdir -p /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny

pwd
#/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes

for file in $(ls */*/*/*.fna.gz); do
dir=$(dirname $file)
cd $dir
gunzip *.fna.gz
cd ../../..
done

#BUSCO - keeping output files
for Genome in $(ls */*/*/*.fna | grep -v 'cds_from_genomic'); do
OutDir=$(dirname $Genome)/BUSCO
Database=/jic/research-groups/Saskia-Hogenhout/BUSCO_sets/v5/hemiptera_odb10
OutFile=$(basename $Genome | sed 's@.fna@@g')_$(echo $Database | cut -d '/' -f7)
ProgDir=~/git_repos/Wrappers/NBI
mkdir $OutDir 
if [ ! -e "$OutDir/hemiptera_odb10/run_hemiptera_odb10/busco_sequences/single_copy_busco_sequences.tar.gz" ]; then
sbatch $ProgDir/run_busco_keep.sh "$Genome" "$Database" "$OutDir" "$OutFile" 
echo "Not done for $Genome - Failed"
else
echo "done for $Genome"
fi
done
#58077709-58072790, 58089658, 58089689, 58089850, 58736464 - 58736698 (medium), 58746250

#Extract complete busco IDs, keep those present in at least 3 genomes:
for file in $(ls */*/*/BUSCO/hemiptera_odb10/run_hemiptera_odb10/full_table.tsv); do
grep -v "^#" $file | awk '$2=="Complete" {print $1}' >> /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/complete_busco_ids.txt;
done

sort /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/complete_busco_ids.txt |uniq -c > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/complete_busco_ids_with_counts.txt
grep -v " 2 " /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/complete_busco_ids_with_counts.txt | grep -v " 1 " > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/complete_busco_ids.txt
awk '{print $2}' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/complete_busco_ids.txt > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/complete_busco_ids_3.txt

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt

for file in $(ls */*/*/BUSCO/hemiptera_odb10/run_hemiptera_odb10/busco_sequences/single_copy_busco_sequences.tar.gz); do
cd $(dirname $file)
tar -xzvf single_copy_busco_sequences.tar.gz
cd /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes
done

#Give unique names to the complete busco genes from each assembly:
for dir in $(ls -d */*/*/BUSCO/hemiptera_odb10/run_hemiptera_odb10/busco_sequences/single_copy_busco_sequences); do
  sppname=$(echo $dir |cut -f1,2,3 -d "/" | sed 's@/@_@g');
  abbrv=$(echo $dir | cut -d '/' -f1 | cut -c 1-3)_$(echo $dir | cut -d '/' -f2 | cut -c 1-3)_$(echo $dir | cut -d '/' -f3)
  echo $sppname
  echo $abbrv
  for file in ${dir}/*.fna; do
    out=$(echo $file |rev |cut -f 1 -d "/"|rev)
    cp $file /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/${sppname}_${out}
    sed -i 's/^>/>'${abbrv}'|/g' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/${sppname}_${out}
  cut -f 1 -d ":" /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/${sppname}_${out} | tr '[:lower:]' '[:upper:]' > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/${sppname}_${out}.1 && mv /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/${sppname}_${out}.1 /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/${sppname}_${out}  
  done
done

#Combine genes from each assembly into a single file per gene:
cd /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt
buscos=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/complete_busco_ids.txt
lines=$(cat $buscos)
for line in $lines; do
  for fna in $(ls *_$line.fna); do
  output=$(echo $line)_nt.fasta
  cat $fna >> $output
  done
done
rm *.fna

#Align the gene sequences;
AlignDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt
for file in $(ls ${AlignDir}/*_nt.fasta); do
OutFile=$(basename $file | sed 's@_nt.fasta@_nt_aligned.fasta@g')
Jobs=$(squeue -u did23faz| grep 'mafft'  | wc -l)
while [ $Jobs -gt 100 ]; do
    sleep 300s
    printf "."
    Jobs=$(squeue -u did23faz| grep 'mafft'| wc -l)
done
OutDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt
ProgDir=~/git_repos/Wrappers/NBI
echo "$file" >> mafft_log.txt
sbatch $ProgDir/sub_mafft_alignment.sh $file $OutDir $OutFile 2>&1 >> mafft_log.txt
done
#58797311

for gene in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/*_aligned.fasta); do
  ID=$(basename $gene |sed 's@_nt_aligned.fasta@@g')
  echo $ID
  mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/$ID
  cp $gene /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/$ID
done

#Trim the alignments:
for Alignment in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/*/*_aligned.fasta); do
  OutDir=$(dirname $Alignment)
  TrimmedName=$(basename $Alignment .fasta)"_trimmed.fasta"
  echo $Alignment
  echo $OutDir
  echo $TrimmedName
  singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/trimal1.4.1.sif trimal -in $Alignment -out $OutDir/$TrimmedName -keepheader -automated1
done

#Diuraphis_noxia biotype_2_v1 has numerous ambiguous nucleotides...

#Trim header names as RAxML need <60 characters in length:
for Alignment in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/*/*_aligned_trimmed.fasta); do
New=$(dirname $Alignment)/$(basename $Alignment .fasta)_edit.fasta
cat $Alignment  | cut -f1 -d '|'  > $New
done

for file in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/*/*_edit.fasta); do
while IFS= read -r line; do
    if [[ "$line" =~ ^\>.+ ]]; then
        echo "$line" | wc -c
    fi
done < $file
done

Jobs=$(squeue -u did23faz| grep 'RAxML'  | wc -l)
while [ $Jobs -gt 100 ]; do
    sleep 300s
    printf "."
    Jobs=$(squeue -u did23faz| grep 'RAxML'| wc -l)
done
#Run RAxML
for Alignment in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/*/*_edit.fasta); do
Prefix=$(basename $Alignment | cut -f1 -d '_')
OutDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/nano_diagnostics/analysis/phylogeny/RAxML/$Prefix
ProgDir=~/git_repos/Wrappers/NBI
mkdir -p $OutDir
Jobs=$(squeue -u did23faz| grep 'RAxML'  | wc -l)
while [ $Jobs -gt 190 ]; do
    sleep 300s
    printf "."
    Jobs=$(squeue -u did23faz| grep 'RAxML'| wc -l)
done
sbatch $ProgDir/run_RAxML_msa.sh $Alignment $OutDir $Prefix
done
#ERROR: $Prefix and $OutDir were the wrong way around resulting in evaluations being deleted, results are GTR+G

#Tested propberly for 9997at7524:
#9997at7524.raxml.log:AIC score: 88062.537364 / AICc score: 88476.378774 / BIC score: 89480.735172
#GTR+G+FO.raxml.log:AIC score: 88062.550680 / AICc score: 88476.392090 / BIC score: 89480.748487
#GTR+G.raxml.log:AIC score: 88062.550680 / AICc score: 88476.392090 / BIC score: 89480.748487
#GTR+R4+FO.raxml.log:AIC score: 88049.098703 / AICc score: 88481.312512 / BIC score: 89490.469678
#GTR.raxml.log:AIC score: 97980.036779 / AICc score: 98390.278538 / BIC score: 99393.599953
#JC+G.raxml.log:AIC score: 90954.927524 / AICc score: 91340.650467 / BIC score: 92336.048264
#JC.raxml.log:AIC score: 100426.479828 / AICc score: 100808.795163 / BIC score: 101802.965936
#Lower is better, score for GTR+G is ~ as good as any other method, therefore not worth retesting all the trees.

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/RAxML

count=0
for gene in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/*/*_edit.fasta); do
ID=$(echo $gene | cut -d '/' -f11)
echo $ID 
x=$(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/nano_diagnostics/analysis/phylogeny/RAxML/${ID}.log)
            if [[ -f ${x} ]]; then
               ((count++))
            fi
mv /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/nano_diagnostics/analysis/phylogeny/RAxML/${ID}* /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/RAxML/.
done
echo "$count"


#Combine individual gene trees into a consensus tree:
source package 0351788b-6639-43fb-8e80-f28600f83cb1
cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/RAxML/*.raxml.bestTree > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/tree-files.txt
ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/RAxML/*.raxml.bootstraps > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/bs-files.txt
astral5 -i /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/tree-files.txt -o /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.astral.tre
astral5 -t 3 -i /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/tree-files.txt -b /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/bs-files.txt -r 100 -o /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstrapped2.astral.tre
tail -n 1 /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstrapped2.astral.tre > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstrapped.consensus2.astral.tre
head -n 100 /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstrapped2.astral.tre > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstraps.astral.tre


astral5 -t 3 -q /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstrapped.consensus2.astral.tre -i /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstraps.astral.tre -o /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstrapped.scored2.astral.tre 
#59183148,59185037,59260253, 59315880 
#no .csv was produced

astral5 -t 3 -i /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/tree-files.txt -q /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstrapped.consensus2.astral.tre -o /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstrapped.localposteriorprobabilityscored2.astral.tre 2> astral-scored.log

/home/theaven/scratch/apps/ASTER/bin/astral4 -i /home/theaven/scratch/apps/ASTER/tree-files.txt -C -c  /home/theaven/scratch/apps/ASTER/hemiptera_phylogeny.bootstrapped.consensus2.astral.tre -o /home/theaven/scratch/apps/ASTER/hemiptera_phylogeny.bootstrapped.consensus2.astral.castle.tre --root API_MEL_GCF_003254395.2 --genelength 1245


mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir
for gene in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/*_aligned.fasta); do
ln -s $gene /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir/.
done

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir2
for gene in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir/*_aligned.fasta); do
name=$(basename $gene)
cat $gene | cut -d '|' -f1 > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir2/${name}
done

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3
for gene in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/*/*_edit.fasta); do
ln -s $gene /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3/.
done

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree2
cd /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree2
AlignDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3/
cpu=4
singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/iqtree_2.3.0.sif iqtree2 -s $AlignDir -m MFP -mtree -T AUTO --threads-max $cpu -b 1000 --verbose
#59196096, 59251934, 59260066

#Not sure if is running correctly, find model seperately:
mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree2-test
cd /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree2-test
AlignDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3/
cpu=32
singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/iqtree_2.3.0.sif iqtree2 -s $AlignDir -m MF -T 32 --threads-max $cpu
#59316762, more cpus: 59322062 - runs out of memory after 4 days only tested 12 models

#Randomly select 100 BUSCO genes and use these for model selection, also select the GTR model as almost certainly best base model
ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3/ | shuf -n 100 | xargs -I {} ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3/{} /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir-100/{}
mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree2-test
cd /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree2-test
AlignDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir-100/
cpu=40
singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/iqtree_2.3.0.sif iqtree2 -s $AlignDir -m MF -T AUTO -mset GTR --threads-max $cpu
#59400442
#Best-fit model: GTR+F+I+R10 chosen according to BIC

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree2
cd /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree2
AlignDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3/
cpu=12
singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/iqtree_2.3.0.sif iqtree2 -s $AlignDir -m GTR+F+I+R10 -B 1000 -T AUTO --threads-max $cpu
#59425290
```
Input data: 161 sequences with 3121748 nucleotide sites
Number of constant sites: 472042 (= 15.1211% of all sites)
Number of invariant (constant or ambiguous constant) sites: 472042 (= 15.1211% of all sites)
Number of parsimony informative sites: 2416927
Number of distinct site patterns: 2748603

Analysis results written to:
  IQ-TREE report:                /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3.iqtree
  Maximum-likelihood tree:       /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3.treefile
  Likelihood distances:          /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3.mldist

Ultrafast bootstrap approximation results written to:
  Split support values:          /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3.splits.nex
  Consensus tree:                /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3.contree
  Screen log file:               /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3.log

```bash
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3* /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree2/.

ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/RAxML/*.raxml.bestTree > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/tree-files2.txt
ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3/*_edit.fasta > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/align-files.txt
cd /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree2
singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/iqtree_2.3.0.sif iqtree2 -t /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3.contree --gcf /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/tree-files.txt --prefix concord -T 1
#Tree with concordance factors written to concord.cf.tree
#Annotated tree (best viewed in FigTree) written to concord.cf.tree.nex
#Tree with branch IDs written to concord.cf.branch
#Concordance factors per branch printed to concord.cf.stat

singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/iqtree_2.3.0.sif iqtree2 -te /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3.contree -p /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3/ --scfl 100 --prefix concord2 -T 8
#59526546, 59645092?

#Partition information was printed to concord2.best_model.nex
#Tree with site concordance factors written to concord2.cf.tree
#Annotated tree (best viewed in FigTree) written to concord2.cf.tree.nex
#Tree with branch IDs written to concord2.cf.branch
#Concordance factors per branch printed to concord2.cf.stat

#Analysis results written to:
#  IQ-TREE report:                concord2.iqtree
#  Maximum-likelihood tree:       concord2.treefile
#  Screen log file:               concord2.log

mv /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/concord2* /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree2/.
mv /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/concord* /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/.


singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/iqtree_2.3.0.sif iqtree2 -t /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstrapped.consensus2.astral.tre --gcf /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/tree-files.txt --prefix concord3 -T 8
singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/iqtree_2.3.0.sif iqtree2 -te /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstrapped.consensus2.astral.tre -p /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3/ --scfl 100 --prefix concord4 -T 8
#59645173
```
For every branch of a reference tree, gCF is defined as the percentage of “decisive” gene trees containing that branch. 

sCF is defined as the percentage of decisive alignment sites supporting a branch in the reference tree.

gCF and sCF complement classical measures of branch support (e.g. bootstrap) in phylogenetics by providing a full description of underlying disagreement among loci and sites.

```bash
#Run RAxML
conda activate raxml
for Alignment in $(ls /mnt/shared/scratch/theaven/uncompressed/hogenhout/phylogeny/9997at7524_nt_aligned_trimmed_edit.fasta); do
Prefix=$(basename $Alignment | cut -f1 -d '_')
OutDir=/mnt/shared/scratch/theaven/uncompressed/hogenhout/phylogeny/RAxML/$Prefix
ProgDir=/home/theaven/scratch/apps/phylogeny
mkdir -p $OutDir
Jobs=$(squeue -u theaven| grep 'RAxML'  | wc -l)
while [ $Jobs -gt 50 ]; do
    sleep 300s
    printf "."
    Jobs=$(squeue -u theaven| grep 'RAxML'| wc -l)
done
sbatch $ProgDir/run_RAxML_msa.sh $Alignment $OutDir $Prefix
done
#19226105
conda deactivate
```

```python
with open("hemip.faa", "r") as f:
    lines = f.readlines()

with open("output.fasta", "w") as f:
    for line in lines:
        if line.startswith(">"):
            line = line.replace("|", "_").replace(" ", "_").replace(".", "_")
        f.write(line)
```
```bash
singularity exec ~/helixer-docker_helixer_v0.3.2_cuda_11.8.0-cudnn8.sif Helixer.py --model-filepath ../databases/helixer/invertebrate_v0.3_a_0600/invertebrate_v0.3_a_0600.h5 --subsequence-length 213840 --overlap-offset 106920 --overlap-core-length 160380 --fasta-path Arabidopsis_lyrata.v.1.0.dna.chromosome.8.fa  \
  --species Arabidopsis_lyrata --gff-output-path Arabidopsis_lyrata_chromosome8_helixer.gff3

```
#### R8s
Making the species tree ultrametric CAFE requires a tree that is ultramatric. There are many ways to obtain ultrametric trees (also known as timetrees, these are phylogenetic trees scaled to time, where all paths from root to tips have the same length). Here, we use a fast program called r8s. You will need to know the number of sites in the alignment used to estimate the species tree (the one you want to make ultrametric), and then you can specify one or more calibration points (ideally, the age or age window of a documented fossil) to scale branch lengths into time units. We provide you with a script that prepares the control file for running r8s on the species tree above (the number of sites is 35157236, and the calibration point for cats and humans is 94). In your shell, type:

```bash
source package /nbi/software/testing/bin/r8s-1.80

echo "#NEXUS" > r8s_ctl_file3.txt
echo "begin trees; " >> r8s_ctl_file3.txt
echo "tree hemiptera_tree = [&R] $(cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/AlignDir3.contree)" >> r8s_ctl_file3.txt
echo "End;" >> r8s_ctl_file3.txt
echo "begin rates;" >> r8s_ctl_file3.txt
echo "blformat nsites=3121748 lengths=persite ultrametric=no;" >> r8s_ctl_file3.txt
echo "collapse;" >> r8s_ctl_file3.txt
#echo "reroot DRO_MEL_GCF_000001215.4;" >> r8s_ctl_file.txt
echo "reroot BOM_MOR_GCF_014905235.1;" >> r8s_ctl_file3.txt
echo "mrca callibration_node_1 THR_PAL_GCA_012932325.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file3.txt
echo "mrca callibration_node_2 BOM_MOR_GCF_014905235.1 ANO_STE_GCF_013141755.1;" >> r8s_ctl_file3.txt
echo "mrca callibration_node_3 BOM_MOR_GCF_014905235.1 TRI_CAS_GCF_000002335.3;" >> r8s_ctl_file3.txt
echo "mrca callibration_node_4 BOM_MOR_GCF_014905235.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file3.txt
echo "mrca callibration_node_5 BAC_COC_GCA_024516035.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file3.txt
echo "mrca callibration_node_6 CIM_LEC_GCA_000648675.3 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file3.txt
echo "mrca callibration_node_7 DAK_VIT_INRAPCF7_V5 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file3.txt
echo "mrca callibration_node_8 PSE_VIB_GCA_033439095.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file3.txt
echo "mrca callibration_node_9 CIM_LEC_GCA_000648675.3 PHI_SPU_GCA_018207615.1;" >> r8s_ctl_file3.txt
echo "mrca callibration_node_10 NIL_LUG_GCA_014356525.1 PHI_SPU_GCA_018207615.1;" >> r8s_ctl_file3.txt
echo "constrain taxon=callibration_node_1 min_age=206.1 max_age=404.6;" >> r8s_ctl_file3.txt
echo "constrain taxon=callibration_node_2 min_age=151.9 max_age=344.7;" >> r8s_ctl_file3.txt
echo "constrain taxon=callibration_node_3 min_age=195 max_age=361.6;" >> r8s_ctl_file3.txt
echo "constrain taxon=callibration_node_4 min_age=330.4 max_age=376;" >> r8s_ctl_file3.txt
echo "constrain taxon=callibration_node_5 min_age=211.9 max_age=351;" >> r8s_ctl_file3.txt
echo "constrain taxon=callibration_node_6 min_age=112.5 max_age=391.7;" >> r8s_ctl_file3.txt
echo "constrain taxon=callibration_node_7 min_age=87.1 max_age=162;" >> r8s_ctl_file3.txt
echo "constrain taxon=callibration_node_8 min_age=168 max_age=286.4;" >> r8s_ctl_file3.txt
echo "constrain taxon=callibration_node_9 min_age=234.9 max_age=366.2;" >> r8s_ctl_file3.txt
echo "constrain taxon=callibration_node_10 min_age=169.6 max_age=344.5;" >> r8s_ctl_file3.txt
echo "divtime method=pl algorithm=tn cvStart=0 cvInc=0.5 cvNum=8 crossv=yes;" >> r8s_ctl_file3.txt
echo "describe plot=chronogram;" >> r8s_ctl_file3.txt
echo "describe plot=tree_description;" >> r8s_ctl_file3.txt
echo "end;" >> r8s_ctl_file3.txt


r8s -b -f r8s_ctl_file.txt > temp_r8s3.txt
tail -n 1 temp_r8s3.txt | cut -c 16- > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree_r8s2.txt

r8s -b -f r8s_ctl_file3.txt > temp_r8s4.txt
tail -n 1 temp_r8s4.txt | cut -c 16- > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree_r8s3.txt

echo "#NEXUS" > r8s_ctl_file22.txt
echo "begin trees; " >> r8s_ctl_file22.txt
echo "tree hemiptera_tree = [&R] $(cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/astralreroot.txt)" >> r8s_ctl_file22.txt
echo "End;" >> r8s_ctl_file22.txt
echo "begin rates;" >> r8s_ctl_file22.txt
echo "blformat nsites=3121748 lengths=persite ultrametric=no;" >> r8s_ctl_file22.txt
echo "collapse;" >> r8s_ctl_file22.txt
#echo "reroot DRO_MEL_GCF_000001215.4;" >> r8s_ctl_file2.txt
#echo "reroot API_MEL_GCF_003254395.2;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_1 THR_PAL_GCA_012932325.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_2 BOM_MOR_GCF_014905235.1 ANO_STE_GCF_013141755.1;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_3 BOM_MOR_GCF_014905235.1 TRI_CAS_GCF_000002335.3;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_4 BOM_MOR_GCF_014905235.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_5 BAC_COC_GCA_024516035.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_6 CIM_LEC_GCA_000648675.3 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_7 DAK_VIT_INRAPCF7_V5 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_8 PSE_VIB_GCA_033439095.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_9 CIM_LEC_GCA_000648675.3 PHI_SPU_GCA_018207615.1;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_10 NIL_LUG_GCA_014356525.1 PHI_SPU_GCA_018207615.1;" >> r8s_ctl_file22.txt
echo "constrain taxon=callibration_node_1 min_age=206.1 max_age=404.6;" >> r8s_ctl_file22.txt
echo "constrain taxon=callibration_node_2 min_age=151.9 max_age=344.7;" >> r8s_ctl_file22.txt
echo "constrain taxon=callibration_node_3 min_age=195 max_age=361.6;" >> r8s_ctl_file22.txt
echo "constrain taxon=callibration_node_4 min_age=330.4 max_age=376;" >> r8s_ctl_file22.txt
echo "constrain taxon=callibration_node_5 min_age=211.9 max_age=351;" >> r8s_ctl_file22.txt
echo "constrain taxon=callibration_node_6 min_age=112.5 max_age=391.7;" >> r8s_ctl_file22.txt
echo "constrain taxon=callibration_node_7 min_age=87.1 max_age=162;" >> r8s_ctl_file22.txt
echo "constrain taxon=callibration_node_8 min_age=168 max_age=286.4;" >> r8s_ctl_file22.txt
echo "constrain taxon=callibration_node_9 min_age=234.9 max_age=366.2;" >> r8s_ctl_file22.txt
echo "constrain taxon=callibration_node_10 min_age=169.6 max_age=344.5;" >> r8s_ctl_file22.txt
echo "divtime method=pl algorithm=tn cvStart=0 cvInc=0.5 cvNum=8 crossv=yes;" >> r8s_ctl_file22.txt
echo "describe plot=chronogram;" >> r8s_ctl_file22.txt
echo "describe plot=tree_description;" >> r8s_ctl_file22.txt
echo "end;" >> r8s_ctl_file22.txt


r8s -b -f r8s_ctl_file22.txt > temp_r8s2.txt
grep 'tree hemiptera_tree' temp_r8s2.txt | cut -c 16- > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/astral_r8s.txt
tail -n 17 temp_r8s2.txt > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/astral_r8s2.txt
#tail -n 1 temp_r8s2.txt | cut -c 16- > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/astral_r8s.txt

r8s -b -f r8s_ctl_file22.txt > temp_r8s22.txt
tail -n 1 temp_r8s22.txt | cut -c 16- > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/astral_r8s2.txt

echo "#NEXUS" > r8s_ctl_file222.txt
echo "begin trees; " >> r8s_ctl_file222.txt
echo "tree hemiptera_tree = [&R] $(cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/iqreroot.txt)" >> r8s_ctl_file222.txt
echo "End;" >> r8s_ctl_file222.txt
echo "begin rates;" >> r8s_ctl_file222.txt
echo "blformat nsites=3121748 lengths=persite ultrametric=no;" >> r8s_ctl_file222.txt
echo "collapse;" >> r8s_ctl_file222.txt
#echo "reroot DRO_MEL_GCF_000001215.4;" >> r8s_ctl_file2.txt
#echo "reroot API_MEL_GCF_003254395.2;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_1 THR_PAL_GCA_012932325.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_2 BOM_MOR_GCF_014905235.1 ANO_STE_GCF_013141755.1;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_3 BOM_MOR_GCF_014905235.1 TRI_CAS_GCF_000002335.3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_4 BOM_MOR_GCF_014905235.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_5 BAC_COC_GCA_024516035.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_6 CIM_LEC_GCA_000648675.3 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_7 DAK_VIT_INRAPCF7_V5 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_8 PSE_VIB_GCA_033439095.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_9 CIM_LEC_GCA_000648675.3 PHI_SPU_GCA_018207615.1;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_10 NIL_LUG_GCA_014356525.1 PHI_SPU_GCA_018207615.1;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_1 min_age=206.1 max_age=404.6;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_2 min_age=151.9 max_age=344.7;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_3 min_age=195 max_age=361.6;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_4 min_age=330.4 max_age=376;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_5 min_age=211.9 max_age=351;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_6 min_age=112.5 max_age=391.7;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_7 min_age=87.1 max_age=162;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_8 min_age=168 max_age=286.4;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_9 min_age=234.9 max_age=366.2;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_10 min_age=169.6 max_age=344.5;" >> r8s_ctl_file222.txt
echo "divtime method=pl algorithm=tn cvStart=0 cvInc=0.5 cvNum=8 crossv=yes;" >> r8s_ctl_file222.txt
echo "describe plot=chronogram;" >> r8s_ctl_file222.txt
echo "describe plot=tree_description;" >> r8s_ctl_file222.txt
echo "end;" >> r8s_ctl_file222.txt

r8s -b -f r8s_ctl_file222.txt > temp_r8s222.txt
tail -n 1 temp_r8s222.txt | cut -c 16- > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/busco_nt/iqtree_r8s222.txt

echo "#NEXUS" > r8s_ctl_file222.txt
echo "begin trees; " >> r8s_ctl_file222.txt
echo "tree hemiptera_tree = [&R] $(cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/hemiptera_phylogeny.bootstrapped.consensus2.astral.castle.tre)" >> r8s_ctl_file222.txt
echo "End;" >> r8s_ctl_file222.txt
echo "begin rates;" >> r8s_ctl_file222.txt
echo "blformat nsites=3121748 lengths=persite ultrametric=no;" >> r8s_ctl_file222.txt
echo "collapse;" >> r8s_ctl_file222.txt
#echo "reroot DRO_MEL_GCF_000001215.4;" >> r8s_ctl_file2.txt
#echo "reroot API_MEL_GCF_003254395.2;" >> r8s_ctl_file22.txt
echo "mrca callibration_node_1 THR_PAL_GCA_012932325.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_2 BOM_MOR_GCF_014905235.1 ANO_STE_GCF_013141755.1;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_3 BOM_MOR_GCF_014905235.1 TRI_CAS_GCF_000002335.3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_4 BOM_MOR_GCF_014905235.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_5 BAC_COC_GCA_024516035.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_6 CIM_LEC_GCA_000648675.3 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_7 DAK_VIT_INRAPCF7_V5 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_8 PSE_VIB_GCA_033439095.1 ACY_PIS_LSR1_AL4_V3;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_9 CIM_LEC_GCA_000648675.3 PHI_SPU_GCA_018207615.1;" >> r8s_ctl_file222.txt
echo "mrca callibration_node_10 NIL_LUG_GCA_014356525.1 PHI_SPU_GCA_018207615.1;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_1 min_age=206.1 max_age=404.6;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_2 min_age=151.9 max_age=344.7;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_3 min_age=195 max_age=361.6;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_4 min_age=330.4 max_age=376;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_5 min_age=211.9 max_age=351;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_6 min_age=112.5 max_age=391.7;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_7 min_age=87.1 max_age=162;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_8 min_age=168 max_age=286.4;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_9 min_age=234.9 max_age=366.2;" >> r8s_ctl_file222.txt
echo "constrain taxon=callibration_node_10 min_age=169.6 max_age=344.5;" >> r8s_ctl_file222.txt
echo "divtime method=pl algorithm=tn cvStart=0 cvInc=0.5 cvNum=8 crossv=yes;" >> r8s_ctl_file222.txt
echo "describe plot=chronogram;" >> r8s_ctl_file222.txt
echo "describe plot=tree_description;" >> r8s_ctl_file222.txt
echo "end;" >> r8s_ctl_file222.txt

r8s -b -f r8s_ctl_file222.txt > temp_r8s-castle.txt

tail -n 1 temp_r8s-castle.txt > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/phylogeny/astral_r8s-castle.txt
```
Running r8s on rerooted trees from ITOL (as the reroot command in r8s doesnt seem to have any effect), this removes the bootstrap values for some reason. R8s non-rerooted trees keep the bootstraps and rerooted trees prior to r8s have bootstrap, I can't be bothered trying to troubleshoot this so am just manually adding the bootstrap values back into the raw newick files.
## Synteny
#### Helixer
```bash
mkdir -p analysis/synteny/helixer

for fasta in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Diaphorina/citri/GCA_030643865.1/GCA_030643865.1_ASM3064386v1_genomic.fna); do
model_filepath=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/helixer/invertebrate_v0.3_m_0100/invertebrate_v0.3_m_0100.h5
lineage=invertebrate
species=$(echo $fasta | cut -d '/' -f8 | cut -c 1)_$(echo $fasta | cut -d '/' -f9)
outfile=$species
outdir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/${species}
ProgDir=~/git_repos/Wrappers/NBI
mkdir $outdir
sbatch $ProgDir/run_helixer.sh $fasta $model_filepath $lineage $species $outfile $outdir
done
#58856194-6, 58857547, 58885166, 58902576, 58967085

grep 'gene' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.gff | wc -l #21,036
grep 'gene' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.gff | wc -l #18,217
grep 'gene' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.gff | wc -l #27,628
```
#### AGAT
```bash
source package 4c883633-af2d-4fac-ab67-a1574f7fe079

Genome=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Diaphorina/citri/GCA_030643865.1/GCA_030643865.1_ASM3064386v1_genomic.fna
Gff=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.gff
OutFile=$(dirname $Gff)/$(basename $Gff | sed 's@.gff@.faa@g')
agat_sp_extract_sequences.pl -g $Gff -f $Genome -t cds --output $OutFile --clean_final_stop --protein

Genome=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Bactericera/cockerelli/GCA_024516035.1/GCA_024516035.1_ASM2451603v1_genomic.fna
Gff=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.gff
OutFile=$(dirname $Gff)/$(basename $Gff | sed 's@.gff@.faa@g')
agat_sp_extract_sequences.pl -g $Gff -f $Genome -t cds --output $OutFile --clean_final_stop --protein

Genome=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Pachypsylla/venusta/GCA_012654025.1/GCA_012654025.1_Pven_dovetail_genomic.fna
Gff=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.gff
OutFile=$(dirname $Gff)/$(basename $Gff | sed 's@.gff@.faa@g')
agat_sp_extract_sequences.pl -g $Gff -f $Genome -t cds --output $OutFile --clean_final_stop --protein

Genome=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/T_apicales_880m_29_3_3.0_0.75_break_TellSeqPurged_curated_nomito_filtered_corrected_softmasked.fa
Gff=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_apicales.gff
OutFile=$(dirname $Gff)/$(basename $Gff | sed 's@.gff@.faa@g')
agat_sp_extract_sequences.pl -g $Gff -f $Genome -t cds --output $OutFile --clean_final_stop --protein

Genome=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/T_anthrisci_820m_48_1_10.0_0.25_break_TellSeqPurged_curated_nomito_filtered_corrected_softmasked.fa
Gff=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_anthrisci.gff
OutFile=$(dirname $Gff)/$(basename $Gff | sed 's@.gff@.faa@g')
agat_sp_extract_sequences.pl -g $Gff -f $Genome -t cds --output $OutFile --clean_final_stop --protein

Genome=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/T_urticae_715m_12_2_3.0_0.5_break_TellSeqPurged_curated_nomito_filtered_corrected_softmasked.fa
Gff=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_urticae.gff
OutFile=$(dirname $Gff)/$(basename $Gff | sed 's@.gff@.faa@g')
agat_sp_extract_sequences.pl -g $Gff -f $Genome -t cds --output $OutFile --clean_final_stop --protein
```
```bash
awk '$3 == "gene" {split($9, parts, "="); printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", $1, $2, $3, $4, $5, $6, $7, $8, parts[2]}' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.gff | awk '{print $1, $9, $4, $5}' | awk 'NF{print "Dc" $0}' > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/Dc.gff
awk '$3 == "gene" {split($9, parts, "="); printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", $1, $2, $3, $4, $5, $6, $7, $8, parts[2]}' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.gff | awk '{print $1, $9, $4, $5}' | awk 'NF{print "Bc" $0}' > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/Bc.gff
awk '$3 == "gene" {split($9, parts, "="); printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", $1, $2, $3, $4, $5, $6, $7, $8, parts[2]}' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.gff | awk '{print $1, $9, $4, $5}' | awk 'NF{print "Pv" $0}' > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/Pv.gff
awk '$3 == "gene" {split($9, parts, "="); printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", $1, $2, $3, $4, $5, $6, $7, $8, parts[2]}' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_apicales.gff | awk '{print $1, $9, $4, $5}' | awk 'NF{print "Tp" $0}' > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/Tp.gff
awk '$3 == "gene" {split($9, parts, "="); printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", $1, $2, $3, $4, $5, $6, $7, $8, parts[2]}' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_anthrisci.gff | awk '{print $1, $9, $4, $5}' | awk 'NF{print "Tn" $0}' > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/Tn.gff
awk '$3 == "gene" {split($9, parts, "="); printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", $1, $2, $3, $4, $5, $6, $7, $8, parts[2]}' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_urticae.gff | awk '{print $1, $9, $4, $5}' | awk 'NF{print "Tu" $0}' > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/Tu.gff

cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/*.gff > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master.gff
#.gff variant names do not have .1 variant information, blast file gene names do
```


```bash
mkdir -p /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB
makeblastdb -in /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.faa -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/D_citri -dbtype prot
makeblastdb -in /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.faa -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/B_cockerelli -dbtype prot
makeblastdb -in /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.faa -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/P_venusta -dbtype prot
makeblastdb -in /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_apicales.faa -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_apicales -dbtype prot
makeblastdb -in /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_anthrisci.faa -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_anthrisci -dbtype prot
makeblastdb -in /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_urticae.faa -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_urticae -dbtype prot

srun -p jic-medium -c 32 --mem 50G --pty bash
source package 37f0ffda-9f66-4391-87e2-38ccd398861d
mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/D_citri -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/B_cockerelli_v_D_citri.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/D_citri -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/P_venusta_v_D_citri.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/D_citri -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_apicales.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_apicales_v_D_citri.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/D_citri -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_anthrisci.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_anthrisci_v_D_citri.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/D_citri -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_urticae.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_urticae_v_D_citri.blast
#59190238

blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/B_cockerelli -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/P_venusta_v_B_cockerelli.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/B_cockerelli -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_apicales.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_apicales_v_B_cockerelli.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/B_cockerelli -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_anthrisci.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_anthrisci_v_B_cockerelli.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/B_cockerelli -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_urticae.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_urticae_v_B_cockerelli.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/B_cockerelli -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/D_citri_v_B_cockerelli.blast
#59190267

blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/P_venusta -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/B_cockerelli_v_P_venusta.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/P_venusta -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_apicales.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_apicales_v_P_venusta.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/P_venusta -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_anthrisci.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_anthrisci_v_P_venusta.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/P_venusta -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_urticae.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_urticae_v_P_venusta.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/P_venusta -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/D_citri_v_P_venusta.blast
#59190270

blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_apicales -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/B_cockerelli_v_T_apicales.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_apicales -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/P_venusta_v_T_apicales.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_apicales -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_anthrisci.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_anthrisci_v_T_apicales.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_apicales -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_urticae.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_urticae_v_T_apicales.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_apicales -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/D_citri_v_T_apicales.blast
#59190272

blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_anthrisci -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/B_cockerelli_v_T_anthrisci.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_anthrisci -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/P_venusta_v_T_anthrisci.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_anthrisci -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_apicales.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_apicales_v_T_anthrisci.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_anthrisci -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_urticae.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_urticae_v_T_anthrisci.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_anthrisci -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/D_citri_v_T_anthrisci.blast
#59190274

blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_urticae -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/B_cockerelli_v_T_urticae.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_urticae -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/P_venusta_v_T_urticae.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_urticae -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_apicales.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_apicales_v_T_urticae.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_urticae -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_anthrisci.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_anthrisci_v_T_urticae.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_urticae -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/D_citri_v_T_urticae.blast
#59190275

blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_urticae -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_urticae.faa -num_threads 32 -evalue 1e-10 -num_alignments 6 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_urticae_v_T_urticae.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_anthrisci -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_anthrisci.faa -num_threads 32 -evalue 1e-10 -num_alignments 6 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_anthrisci_v_T_anthrisci.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/T_apicales -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_apicales.faa -num_threads 32 -evalue 1e-10 -num_alignments 6 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_apicales_v_T_apicales.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/P_venusta -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.faa -num_threads 32 -evalue 1e-10 -num_alignments 6 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/P_venusta_v_P_venusta.blast
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/B_cockerelli -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.faa -num_threads 32 -evalue 1e-10 -num_alignments 6 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/B_cockerelli_v_B_cockerelli.blast 
blastp -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/ncbiDB/D_citri -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.faa -num_threads 32 -evalue 1e-10 -num_alignments 5 -outfmt 6 -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/D_citri_v_D_citri.blast
#59196288

cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/*.blast | sed 's/.1_helixer//g' > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master.blast
awk 'BEGIN {FS=OFS="\t"} {sub(/\.1$/, "", $1); sub(/\.1$/, "", $2); print}' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master.blast > temp.blast && mv temp.blast /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master.blast
< /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master.blast tr ' ' '\t' > temp.blast && mv temp.blast /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master.blast
< /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master.gff tr ' ' '\t' > temp.gff && mv temp.gff /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master.gff

source package 038f5eb6-dc79-46b5-bb52-a86ed67aa64a
MCScanX /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master

#1180980 matches imported (1343327 discarded)
#188870 pairwise comparisons
#

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/tapitant
cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_anthrisci_v_T_anthrisci.blast /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_apicales_v_T_apicales.blast /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_apicales_v_T_anthrisci.blast /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_anthrisci_v_T_apicales.blast > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/tapitant/master.blast
cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/Tn.gff /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/Tp.gff > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/tapitant/master.gff
awk 'BEGIN {FS=OFS="\t"} {sub(/\.1$/, "", $1); sub(/\.1$/, "", $2); print}' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/tapitant/master.blast > temp.blast && mv temp.blast /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/tapitant/master.blast
< /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/tapitant/master.blast tr ' ' '\t' > temp.blast && mv temp.blast /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/tapitant/master.blast
< /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/tapitant/master.gff tr ' ' '\t' > temp.gff && mv temp.gff /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/tapitant/master.gff
MCScanX /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/tapitant/master
```
```python
from Bio import SeqIO

# Path to your multi-FASTA file
fasta_file = "/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Diaphorina/citri/GCA_030643865.1/GCA_030643865.1_ASM3064386v1_genomic.fna"
fasta_file = "/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Bactericera/cockerelli/GCA_024516035.1/GCA_024516035.1_ASM2451603v1_genomic.fna"
fasta_file = "/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Genomes/Pachypsylla/venusta/GCA_012654025.1/GCA_012654025.1_Pven_dovetail_genomic.fna"
fasta_file = "/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/T_apicales_880m_29_3_3.0_0.75_break_TellSeqPurged_curated_nomito_filtered_corrected_softmasked.fa"
fasta_file = "/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/T_anthrisci_820m_48_1_10.0_0.25_break_TellSeqPurged_curated_nomito_filtered_corrected_softmasked.fa"
fasta_file = "/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/T_urticae_715m_12_2_3.0_0.5_break_TellSeqPurged_curated_nomito_filtered_corrected_softmasked.fa"

# Iterate over each sequence in the multi-FASTA file
for record in SeqIO.parse(fasta_file, "fasta"):
    # Print the ID and length of each sequence
    print(f"Contig ID: {record.id}, Length: {len(record.seq)}")

    seqkit fx2tab -l /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/T_anthrisci_820m_48_1_10.0_0.25_break_TellSeqPurged_curated_nomito_filtered_corrected_softmasked.fa | awk '{print $1 "\t" length($2)}'
```
```bash
awk -F'\t' '{print $1}' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master.gff | grep '^Tp' | sort -u

cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master.blast /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.blast
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master.gff /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff

sed -i 's/TuSUPER_1/Tu1/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_2/Tu2/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_3/Tu3/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_4/Tu4/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_5/Tu5/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_6/Tu6/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_7/Tu7/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_8/Tu8/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_9/Tu9/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_10/Tu10/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_11/Tu11/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_12/Tu12/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TuSUPER_13/Tu13/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff

sed -i 's/TnSUPER_1/Dp1/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_2/Dp2/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_3/Dp3/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_4/Dp4/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_5/Dp5/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_6/Dp6/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_7/Dp7/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_8/Dp8/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_9/Dp9/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_10/Dp10/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_11/Dp11/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_12/Dp12/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TnSUPER_13/Dp13/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff

sed -i 's/TpSUPER_1/Da1/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TpSUPER_2/Da2/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TpSUPER_3/Da3/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TpSUPER_4/Da4/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TpSUPER_5/Da5/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TpSUPER_6/Da6/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TpSUPER_7/Da7/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TpSUPER_8/Da8/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TpSUPER_9/Da9/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TpSUPER_10/Da10/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's@TpSUPER_11_1@Da11@' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TpSUPER_12/Da12/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/TpSUPER_13/Da13/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/Da11_1/Da11/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff

sed -i 's/BcCM044955.1/Bc1/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044956.1/Bc2/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044957.1/Bc3/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044958.1/Bc4/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044959.1/Bc5/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044960.1/Bc6/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044961.1/Bc7/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044962.1/Bc8/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044963.1/Bc9/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044964.1/Bc10/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044965.1/Bc11/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044966.1/Bc12/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/BcCM044967.1/Bc13/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff

sed -i 's/DcCP125321.1/Dc9/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125322.1/Dc2/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125323.1/Dc12/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125324.1/Dc7/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125325.1/Dc5/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125326.1/Dc13/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125327.1/Dc11/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125328.1/Dc8/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125329.1/Dc6/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125330.1/Dc10/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125331.1/Dc4/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125332.1/Dc3/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/DcCP125333.1/Dc1/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff

sed -i 's/PvCM022874.1/Pv1/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvCM022875.1/Pv2/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvCM022876.1/Pv3/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvCM022877.1/Pv4/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvCM022878.1/Pv5/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvCM022879.1/Pv6/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvCM022880.1/Pv7/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvCM022881.1/Pv8/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvCM022882.1/Pv9/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvCM022883.1/Pv10/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvCM022884.1/Pv11/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvCM022885.1/Pv12/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff
sed -i 's/PvJAAEEI010039819.1/Pv13/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff

sed -i 's/.1_helixer//' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.blast
sed -i 's/.1_helixer//' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff

grep 'apicales' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.blast | awk '{print $1}' | grep 'anthrisci'
grep 'anthrisci' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.blast | awk '{print $1}' | grep 'apicales'

grep 'apicales' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff | awk '{print $1}' | grep 'anthrisci'
grep 'anthrisci' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited.gff | awk '{print $1}' | grep 'apicales'

source package 038f5eb6-dc79-46b5-bb52-a86ed67aa64a
MCScanX /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/master/master-edited

#1,208,756 matches imported (1,478,439 discarded)
#62,617 pairwise comparisons
```

```bash
awk '$1 ~ /^T_apicales_SUPER_4/ && $2 ~ /^T_anthrisci_SUPER_9/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_apicales_v_T_anthrisci.blast
awk '$1 ~ /^T_apicales_SUPER_9/ && $2 ~ /^T_anthrisci_SUPER_4/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_apicales_v_T_anthrisci.blast

awk '$1 ~ /^T_anthrisci_SUPER_4/ && $2 ~ /^T_apicales_SUPER_9/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_anthrisci_v_T_apicales.blast
awk '$1 ~ /^T_anthrisci_SUPER_9/ && $2 ~ /^T_apicales_SUPER_4/' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/mcscanx/intermediateData/T_anthrisci_v_T_apicales.blast
```
for pkg in docker.io docker-doc docker-compose docker-compose-v2 podman-docker containerd runc; do apt-get remove $pkg; done

#### Genespace
```bash
source package 03380c15-2730-4b19-b17a-5a435e152681

mkdir -p analysis/synteny/genespace/raw_genomes/D_citri
mkdir -p analysis/synteny/genespace/raw_genomes/T_apicales
mkdir -p analysis/synteny/genespace/raw_genomes/T_anthrisci
mkdir -p analysis/synteny/genespace/raw_genomes/T_urticae # highly fragmented
mkdir -p analysis/synteny/genespace/raw_genomes/B_cockerelli
mkdir -p analysis/synteny/genespace/raw_genomes/P_venusta

ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.gff analysis/synteny/genespace/raw_genomes/B_cockerelli/.
ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/B_cockerelli/B_cockerelli.faa analysis/synteny/genespace/raw_genomes/B_cockerelli/.

ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.gff analysis/synteny/genespace/raw_genomes/P_venusta/.
ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/P_venusta/P_venusta.faa analysis/synteny/genespace/raw_genomes/P_venusta/.

ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.gff analysis/synteny/genespace/raw_genomes/D_citri/.
ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/helixer/D_citri/D_citri.faa analysis/synteny/genespace/raw_genomes/D_citri/.

ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_anthrisci.gff analysis/synteny/genespace/raw_genomes/T_anthrisci/.
ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_anthrisci.faa analysis/synteny/genespace/raw_genomes/T_anthrisci/.

ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_apicales.gff analysis/synteny/genespace/raw_genomes/T_apicales/.
ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/helixer/T_apicales.faa analysis/synteny/genespace/raw_genomes/T_apicales/.

ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/filtered/purge_dups/purge_haplotigs/break10x/yahs/filtered/inspector/repeatmasker/softmask/helixer/T_urticae.gff analysis/synteny/genespace/raw_genomes/T_urticae/.
ln -s /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/filtered/purge_dups/purge_haplotigs/break10x/yahs/filtered/inspector/repeatmasker/softmask/helixer/T_urticae.faa analysis/synteny/genespace/raw_genomes/T_urticae/.

for file in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/genespace/raw_genomes/*/*.faa); do
Out=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/genespace/peptide/$(basename $file | sed 's@.faa@.fa@g')
cat $file | cut -d ' ' -f1 > $Out
done

for file in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/genespace/raw_genomes/*/*.gff); do
Out=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/genespace/bed/$(basename $file | sed 's@.gff@.bed@g')
cat "$file" | awk '$3 == "gene"' | cut -f1,4,5,9 | awk -F'\t' -v OFS='\t' '{ $4 = $4 ".1"; gsub("ID=", "", $4); print }' > $Out
done

source package 038f5eb6-dc79-46b5-bb52-a86ed67aa64a
source package 03380c15-2730-4b19-b17a-5a435e152681
R
```
```R
library(GENESPACE)
gpar <- init_genespace(
  wd = "/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/synteny/genespace", 
  path2mcscanx = "/software/03380c15-2730-4b19-b17a-5a435e152681/bin/MCScanX/")
gpar <- run_genespace(gsParam = gpar)
```
>D_citri_KI478532.1_000002.1
D_citri_KI472552.1_000002.1
>D_citri_KI474893.1_000002.1

#### MEGAN
We modified a previous pipeline (Nikoh et al., 2010) to detect genes of bacterial origin expressed in our RNA-seq data. First, the transcriptome assemblies from both RNA-seq samples were searched by BLASTX v2.2.25+ (-evalue 1 × 10−3 -outfmt 7) against the nr database (posted January 2012) and the blast results were visualized in Megan v4 (Huson et al., 2011) as metatranscriptomic data. By plotting number of sequences assigned to distinct taxonomic units, we identified several HGT candidates and contaminant bacteria.

To identify psyllid genes of bacterial origin, the output from BlastX searches against the NCBI nr database for each transcript was analyzed with MEGAN v4.70.4 (Huson et al. 2011) with the following parameters: min support = 5; min score = 50; top percent = 10; win score = 0; and min complexity = 0.3.
```bash
source package d6092385-3a81-49d9-b044-8ffb85d0c446
mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/blast
blastp -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/braker.aa -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/blast/nr_08042024/nr -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/blast/HGT_results.out -evalue 1e-10 -outfmt 6 -num_threads 32

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/blast
blastp -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/braker.aa -db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/blast/nr_08042024/nr -out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/blast/HGT_results.out -evalue 1e-10 -outfmt 6 -num_threads 32


source package dd43df1f-7eb2-4011-88fc-c457e801ddd0
cd /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/diamond/nr_03072024
diamond makedb --in nr.gz -d /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/diamond/nr_03072024/diamond-nr --taxonmap prot.accession2taxid.gz --taxonnodes nodes.dmp --taxonnames names.dmp --threads 64
#1338398

source package dd43df1f-7eb2-4011-88fc-c457e801ddd0
mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond
diamond blastp --threads 32 --query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/T_anthrisci_braker3+helixer+tsebra_loose_longest.faa --db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/diamond/nr_03072024/nr --outfmt 6 --out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond/HGT_results2.out --evalue 1e-10 --daa
#1245703, 1245979

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond
diamond blastp --threads 32 --query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/T_apicalis_braker3+helixer+tsebra_loose_longest.faa --db  /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/diamond/nr_03072024/nr --outfmt 6 --out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond/HGT_results2.out --evalue 1e-10 --daa
#1245709, 1245981

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+loose/diamond
diamond blastp --threads 32 --query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+loose/braker_longest.faa --db  /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/diamond/nr_03072024/nr --outfmt 6 --out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+loose/diamond/HGT_results2.out --evalue 1e-10 --daa
#1271631

source package dd43df1f-7eb2-4011-88fc-c457e801ddd0
mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond
diamond blastp --threads 64 --query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/T_anthrisci_braker3+helixer+tsebra_loose_longest.faa --db /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/diamond/nr_03072024/diamond-nr -f 100 --out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond/HGT_results2.daa --evalue 1e-10
#1338422

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond
diamond blastp --threads 64 --query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/T_apicalis_braker3+helixer+tsebra_loose_longest.faa --db  /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/diamond/nr_03072024/diamond-nr  -f 100 --out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond/HGT_results2.daa --evalue 1e-10
#1338421

mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+loose/diamond
diamond blastp --threads 64 --query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+loose/braker_longest.faa --db  /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/diamond/nr_03072024/diamond-nr  -f 100 --out /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+loose/diamond/HGT_results2.daa --evalue 1e-10
#1338419


source package f2222b6a-caea-4445-a268-b9f1a92c5602
srun -p jic-medium  -c 8 --mem 32G --pty bash
singularity exec ../containers/megan_6.25.9--h9ee0642_0 daa-meganizer -i /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+loose/diamond/HGT_results2.daa -mdb /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/megan/megan-map-Feb2022.db --threads 8
singularity exec ../containers/megan_6.25.9--h9ee0642_0 daa-meganizer -i /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond/HGT_results2.daa -mdb /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/megan/megan-map-Feb2022.db --threads 8
singularity exec ../containers/megan_6.25.9--h9ee0642_0 daa-meganizer -i /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond/HGT_results2.daa -mdb /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/databases/megan/megan-map-Feb2022.db --threads 8

cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond/HGT_results2.daa ./T_anthrisci_HGT_results2.daa
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/diamond/HGT_results2.daa ./T_apicales_HGT_results2.daa
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+loose/diamond/HGT_results2.daa ./T_urticae_HGT_results2.daa
T_urticae_HGT_results2.daa
```
```bash
source package d6092385-3a81-49d9-b044-8ffb85d0c446
cat podosporaanserina_GCF_000226545.1_ASM22654v1_protein.faa neurosporacrassa_GCF_000182925.2_NC12_protein.faa > mydb.faa
mkdir db

ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/HTG.fa

makeblastdb -in /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/braker.aa -dbtype prot -title apidb -out analysis/hgt/db/api
makeblastdb -in /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/braker.aa -dbtype prot -title antdb -out analysis/hgt/db/ant
makeblastdb -in /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+loose/braker.aa -dbtype prot -title urtdb -out analysis/hgt/db/urt

blastx -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/HTG.fa -db analysis/hgt/db/api -max_target_seqs 5 -out analysis/hgt/api-blastx-results.out -outfmt 6 -evalue 1e-5
blastx -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/HTG.fa -db analysis/hgt/db/ant -max_target_seqs 5 -out analysis/hgt/ant-blastx-results.out -outfmt 6 -evalue 1e-5
blastx -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/HTG.fa -db analysis/hgt/db/urt -max_target_seqs 5 -out analysis/hgt/urt-blastx-results.out -outfmt 6 -evalue 1e-5

cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/T_apicales_880m_29_3_3.0_0.75_break_TellSeqPurged_curated_nomito_filtered_corrected.fa | sed 's@>@>Tapi_@g' >> analysis/hgt/db/tapitantturt.fa 
cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/T_anthrisci_820m_48_1_10.0_0.25_break_TellSeqPurged_curated_nomito_filtered_corrected.fa | sed 's@>@>Tant_@g' >> analysis/hgt/db/tapitantturt.fa 
cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/T_urticae_715m_12_2_3.0_0.5_break_TellSeqPurged_curated_nomito_filtered_corrected.fa | sed 's@>@>Turt_@g' >> analysis/hgt/db/tapitantturt.fa
makeblastdb -in /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/hgt/db/tapitantturt.fa -dbtype nucl -title ttt -out analysis/hgt/db/ttt

blastn -query /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/HTG.fa -db analysis/hgt/db/ttt -max_target_seqs 9999 -out analysis/hgt/ttt-blastn-results.out -outfmt 6 -evalue 1e-5



singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/python3.sif python3 ~/git_repos/Scripts/NBI/seq_get.py --id_file temp.txt --input /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/braker.aa --output analysis/hgt/Clostridioides-difficile.faa
```
#### Carsonella
```bash
REFERENCE="/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/Candidatus/Carsonella/ruddii/GCA_002009355.1/GCA_002009355.1_ASM200935v1_genomic.fna "
VCF="/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/resequencing_psyllids/snp_calling/Carsonella/ruddii/variants_filtered.recode.vcf.gz"
samples=$(bcftools query -l $VCF)
for sample in $samples; do
    ID=$(echo $sample | sed 's@_trimmed_1_sorted_MarkDups@@g')
    mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/Candidatus/Carsonella/ruddii/${ID}
    bcftools view --threads 8 -s $sample -Oz -o ${sample}.vcf.gz $VCF
    bcftools index --threads 8 ${sample}.vcf.gz
    cat $REFERENCE | bcftools consensus ${sample}.vcf.gz > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/Candidatus/Carsonella/ruddii/${ID}/${sample}_consensus.fasta
    rm ${sample}.vcf.gz ${sample}.vcf.gz.csi
done

for Genome in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/Candidatus/Carsonella/ruddii/*/*.fasta /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/Candidatus/Carsonella/ruddii/*/GCA*.fna /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/Candidatus/Carsonella/ruddii/*/GCF*.fna /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/Candidatus/Carsonella/ruddii/*/*.fa); do
OutDir=$(dirname $Genome)/prokka
OutFile=$(echo $Genome | cut -d '/' -f11)
Locustag=$(echo $Genome | cut -d '/' -f11)
mkdir $OutDir
ProgDir=~/git_repos/Wrappers/NBI
echo $Locustag
sbatch $ProgDir/run_prokka.sh $Genome $OutDir $OutFile $Locustag
done
#63459775-63459844

ProjDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts
cd $ProjDir
IsolateAbrv=All_carsonella_1
WorkDir=analysis/orthology/orthofinder/$IsolateAbrv
mkdir -p $WorkDir
mkdir -p $WorkDir/formatted
mkdir -p $WorkDir/goodProteins
mkdir -p $WorkDir/badProteins  
cd $WorkDir/formatted

x=0
for Fasta_file in /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/Candidatus/Carsonella/ruddii/*/prokka/*.faa; do
    x=$(printf "%02d" $x)
    Id_field=1
    Taxon_code=$(echo $Fasta_file | cut -d '/' -f11)
    echo $x
    echo $Taxon_code
    ~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl $Taxon_code $Fasta_file $Id_field
    x=$((10#$x + 1))
done

cd $ProjDir

for WorkDir in $(ls -d analysis/orthology/orthofinder/All_carsonella_1); do
Input_dir=$WorkDir/formatted
Min_length=10
Max_percent_stops=20
Good_proteins_file=$WorkDir/goodProteins/goodProteins.fasta
Poor_proteins_file=$WorkDir/badProteins/poorProteins.fasta
~/git_repos/Scripts/NBI/orthomclFilterFasta.pl $Input_dir $Min_length $Max_percent_stops $Good_proteins_file $Poor_proteins_file
done

Prefix=All_carsonella_1
OutDir=orthofinder52000
sbatch ~/git_repos/Wrappers/NBI/run_orthofinder.sh $Input_dir $Prefix $OutDir
#63460245, 63460646

nano ../Symbionts/analysis/orthology/orthofinder/All_carsonella_1/formatted/orthofinder52000/Results_All_carsonella_1/WorkingDirectory/Alignments_ids/SpeciesTreeAlignment.fa
nano ../Symbionts/analysis/orthology/orthofinder/All_carsonella_1/formatted/orthofinder52000/Results_All_carsonella_1/Log.txt

mkdir -p /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/phylogeny/orthofinder/iqtree2
cd /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/phylogeny/orthofinder/iqtree2
Align=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/orthology/orthofinder/All_carsonella_1/formatted/orthofinder52000/Results_All_carsonella_1/WorkingDirectory/Alignments_ids/SpeciesTreeAlignment.fa
cpu=32
singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/iqtree_2.3.0.sif iqtree2 -s $Align -m MFA -T AUTO --threads-max $cpu
#63466874

cd /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/phylogeny/orthofinder/iqtree2
Align=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/orthology/orthofinder/All_carsonella_1/formatted/orthofinder52000/Results_All_carsonella_1/WorkingDirectory/Alignments_ids/SpeciesTreeAlignment.fa
cpu=4
singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/iqtree_2.3.0.sif iqtree2 -s $Align -m mtInv+F+I+R4 -B 1000 -T AUTO --threads-max $cpu -redo 
#6429332

#Analysis results written to:
#  IQ-TREE report:                /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/orthology/orthofinder/All_carsonella_1/formatted/orthofinder52000/Results_All_carsonella_1/WorkingDirectory/Alignments_ids/SpeciesTreeAlignment.fa.iqtree
#  Maximum-likelihood tree:       /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/orthology/orthofinder/All_carsonella_1/formatted/orthofinder52000/Results_All_carsonella_1/WorkingDirectory/Alignments_ids/SpeciesTreeAlignment.fa.treefile
#  Likelihood distances:          /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/orthology/orthofinder/All_carsonella_1/formatted/orthofinder52000/Results_All_carsonella_1/WorkingDirectory/Alignments_ids/SpeciesTreeAlignment.fa.mldist

#Ultrafast bootstrap approximation results written to:
#  Split support values:          /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/orthology/orthofinder/All_carsonella_1/formatted/orthofinder52000/Results_All_carsonella_1/WorkingDirectory/Alignments_ids/SpeciesTreeAlignment.fa.splits.nex
#  Consensus tree:                /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/orthology/orthofinder/All_carsonella_1/formatted/orthofinder52000/Results_All_carsonella_1/WorkingDirectory/Alignments_ids/SpeciesTreeAlignment.fa.contree
#  Screen log file:               /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/orthology/orthofinder/All_carsonella_1/formatted/orthofinder52000/Results_All_carsonella_1/WorkingDirectory/Alignments_ids/SpeciesTreeAlignment.fa.log

ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Symbionts/analysis/phylogeny/orthofinder/iqtree2/Carsonella-iqtree.tree.txt
  ```
#### OrthoFinder
```bash
for file in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/*.faa); do
singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/python3.sif python3 ~/git_repos/Scripts/NBI/longest_variant.py $file $(dirname $file)/$(basename $file | sed 's@.faa@_longest.faa@g')
done


ProjDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae
cd $ProjDir
IsolateAbrv=23_hemiptera-4
WorkDir=analysis/orthofinder/$IsolateAbrv
mkdir -p $WorkDir
mkdir -p $WorkDir/formatted
mkdir -p $WorkDir/goodProteins
mkdir -p $WorkDir/badProteins  

Id_field=1

for file in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/*_longest.faa | grep -v 'Scch'); do
Taxon_code=$(basename $file | cut -d '_' -f1)
Fasta_file=$file
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
done

Taxon_code=Acpi
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Acyrthosiphon_pisum_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Apgl
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Aphis_glycines_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Aplu
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Apolygus_lucorum_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Baco
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Bactericera_cockerelli_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Beta
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Bemisia_tabaci_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Cile
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Cimex_lectularius_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Dici
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Diaphorina_citri_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Dyap
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Dyspera_apicales_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Dypa
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Dyspera_pallida_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Froc
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Frankliniella_occidentalis_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Haha
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Halyomorpha_halys_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Hoco
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Hormaphis_cornu_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Maqu
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Macrosteles_quadrilineatus_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Mype
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Myzus_persicae_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Nete
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Nesidiocoris_tenuis_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Nevi
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Nezara_viridula_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Nilu
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Nilaparvata_lugens_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Pave
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Pachypsylla_venusta_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Plci
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Planococcus_citri_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Ripe
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Riptortus_pedestris_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Sifl
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Sipha_flava_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Siav
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Sitobion_avenae_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta
Taxon_code=Trur
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Trioza_urticae_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta

Taxon_code=Drme
Fasta_file=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Drosophila_melanogaster_longest.faa
~/git_repos/Scripts/NBI/orthomclAdjustFasta.pl  $Taxon_code $Fasta_file $Id_field
mv "$Taxon_code".fasta $WorkDir/formatted/"$Taxon_code".fasta

for WorkDir in $(ls -d analysis/orthofinder/$IsolateAbrv); do
Input_dir=$WorkDir/formatted
Min_length=10
Max_percent_stops=20
Good_proteins_file=$WorkDir/goodProteins/goodProteins.fasta
Poor_proteins_file=$WorkDir/badProteins/poorProteins.fasta
~/git_repos/Scripts/NBI/orthomclFilterFasta.pl $Input_dir $Min_length $Max_percent_stops $Good_proteins_file $Poor_proteins_file
done

Prefix=23_hemiptera
OutDir=orthofinder52000
sbatch ~/git_repos/Wrappers/NBI/run_orthofinder.sh $Input_dir $Prefix $OutDir
#1415420, 1495869, 1554986, 1554988

#reroot the tree in itol:
nano /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23_hemiptera/formatted/orthofinder52000-2/Results_23_hemiptera/Species_Tree/SpeciesTree_rooted-rerooted.txt
nano /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23_hemiptera-2/formatted/orthofinder52000-2/Results_23_hemiptera/Species_Tree/SpeciesTree_rooted-rerooted.txt
nano /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/collapsed-tree-itol.txt #reroot + prune the AlignDir3.treefile to get desired species and then use python to Collapse nodes in the tree that have only one child.

#count the sites in the alignment used to make the tree:
seqkit fx2tab -l /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23_hemiptera/formatted/orthofinder52000/Results_23_hemiptera/MultipleSequenceAlignments/SpeciesTreeAlignment.fa | awk '{print $1 "\t" length($2)}' #87992
seqkit fx2tab -l /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23_hemiptera-2/formatted/orthofinder52000-2/Results_23_hemiptera/MultipleSequenceAlignments/SpeciesTreeAlignment.fa | awk '{print $1 "\t" length($2)}' #60279
nano /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/collapsed-tree-itol.txt #iqtree reports 2748603 distinct site patterns were used in making the tree

source package /nbi/software/testing/bin/r8s-1.80
echo "#NEXUS" > r8s_ctl_file-m.txt
echo "begin trees;" >> r8s_ctl_file-m.txt
echo "tree hemiptera_tree = [&R] $(cat /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/collapsed-tree-itol.txt)" >> r8s_ctl_file-m.txt
echo "end;" >> r8s_ctl_file-m.txt
echo "begin rates;" >> r8s_ctl_file-m.txt
echo "blformat nsites=2748603 lengths=persite ultrametric=no;" >> r8s_ctl_file-m.txt
echo "collapse;" >> r8s_ctl_file-m.txt
echo "mrca callibration_node_1 Mype Froc;" >> r8s_ctl_file-m.txt
echo "mrca callibration_node_2 Mype Apgl;" >> r8s_ctl_file-m.txt
echo "mrca callibration_node_3 Mype Nilu;" >> r8s_ctl_file-m.txt
echo "mrca callibration_node_4 Mype Plci;" >> r8s_ctl_file-m.txt
#echo "mrca callibration_node_5 Mype Drme;" >> r8s_ctl_file-m.txt
echo "constrain taxon=callibration_node_1 min_age=206 max_age=404.6;" >> r8s_ctl_file-m.txt
echo "constrain taxon=callibration_node_2 min_age=23.0 max_age=64.5;" >> r8s_ctl_file-m.txt
echo "constrain taxon=callibration_node_3 min_age=112.5 max_age=391.7;" >> r8s_ctl_file-m.txt
echo "constrain taxon=callibration_node_4 min_age=168.0 max_age=286.4;" >> r8s_ctl_file-m.txt
#echo "constrain taxon=callibration_node_5 min_age=330.4 max_age=376.0;" >> r8s_ctl_file-m.txt
echo "divtime method=pl algorithm=tn cvStart=0 cvInc=0.5 cvNum=8 crossv=yes;" >> r8s_ctl_file-m.txt
echo "describe plot=chronogram;" >> r8s_ctl_file-m.txt
echo "describe plot=tree_description;" >> r8s_ctl_file-m.txt
echo "end;" >> r8s_ctl_file-m.txt

r8s -b -f r8s_ctl_file-m.txt > temp_r8s-m.txt
tail -n 1 temp_r8s-m.txt | cut -c 16- > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23_hemiptera/formatted/orthofinder52000/Results_23_hemiptera/Species_Tree/SpeciesTree_rooted_r8s.txt
tail -n 1 temp_r8s-m.txt | cut -c 16- > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23_hemiptera-2/formatted/orthofinder52000-2/Results_23_hemiptera/Species_Tree/SpeciesTree_rooted_r8s.txt
tail -n 1 temp_r8s-m.txt | cut -c 16- > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23-collapsed-tree-itol-r8s.txt

#convert orthofinder orthogroup counts to CAFE format
mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23_hemiptera-3/formatted/orthofinder52000/Results_23_hemiptera/Orthogroups/Orthogroups.GeneCount.tsv /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/.
awk 'BEGIN { FS=OFS="\t" } { print "n/a", $0 }' /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/Orthogroups.GeneCount.tsv | awk 'BEGIN { FS=OFS="\t" } NR==1 { sub(/.*/, "Desc", $1) }; { print $0 }' | awk 'BEGIN { FS=OFS="\t" } { NF--; print }' | awk 'BEGIN { FS=OFS="\t" } { if (NR == 1) { print; next } for(i=3; i<=NF; i++) if ($i > 99) next; print }' > tmpfile && mv tmpfile /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/Orthogroups.GeneCount.tsv

OutDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe
OutFile=22_hemiptera+Froc
Orthogroups=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/Orthogroups.GeneCount.tsv
Tree=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23-collapsed-tree-itol-r8s.txt
lamdaTree=NA
sbatch ~/git_repos/Wrappers/NBI/run_cafe5.sh $OutDir $OutFile $Orthogroups $Tree $lamdaTree
#1476224, 1477667, 1480369, 1480632, 1543821, 1544020, 1608651
#Filtering families not present at the root from: 33158 to 7185 - Drosophila as root
#Filtering families not present at the root from: 31451 to 8178 - Frankliniella as root

singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/python3.sif python3 ~/git_repos/Scripts/NBI/cafe5_draw_tree.py -i /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/23_hemiptera_results/Base_clade_results.txt -d /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/23_hemiptera_results/Base_report.cafe -o /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/23_hemiptera_results/clade_results.png

singularity exec --overlay /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/overlays/cafe-plot.img:ro /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/python3.sif cafeplotter -i /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/23_hemiptera_results -o /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/23_hemiptera_results

singularity exec --overlay /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/overlays/cafe-plot.img:ro /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/python3.sif python3 ~/git_repos/Scripts/NBI/CAFE_fig.py /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/23_hemiptera_results/Base_report.cafe


mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/fastortho
for fasta in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/*_longest.faa); do
Species=$(basename $fasta | sed 's@_longest.faa@@g' | sed 's@_@ @g')
sed '/^>/ s/$/ \['"$Species"'\]/' $fasta >> /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/fastortho/mixed_genome_fasta2.faa
done

OutDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/fastortho
OutFile="23-hemiptera-2"
Fasta=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/fastortho/mixed_genome_fasta2.faa
Evalue="1-e2"
sbatch ~/git_repos/Wrappers/NBI/run_fastortho.sh $OutDir $OutFile $Fasta $Evalue
#1491562, 1495688

singularity exec --overlay /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/overlays/cafe-plot.img:ro /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/python3.sif cafeplotter -i /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/23_hemiptera_results/result_summary.tsv -o /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/test.png
```
```bash
orthofinder_results=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23_hemiptera-3/formatted/orthofinder52000/Results_23_hemiptera/Orthogroups/Orthogroups.tsv

#All
for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
x=$(grep 'Acpi' $orthofinder_results | grep 'Apgl' | grep 'Aplu' | grep 'Baco' | grep 'Beta' | grep 'Cile' | grep 'Dici' | grep 'Dyap' | grep 'Dypa' | grep 'Froc' | grep 'Scch' | grep 'Haha' | grep 'Maqu' | grep 'Mype' | grep 'Nete' | grep 'Nevi' | grep 'Nilu' | grep 'Pave' | grep 'Plci' | grep 'Ripe' | grep 'Siav' | grep 'Sifl' | grep 'Trur' | tr '\t' '\n' | grep -E "^$species" | tr ',' '\n' | sort | uniq | wc -l)
echo $species : $x
done

#hemiptera
for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
x=$(grep 'Acpi' $orthofinder_results | grep 'Apgl' | grep 'Aplu' | grep 'Baco' | grep 'Beta' | grep 'Cile' | grep 'Dici' | grep 'Dyap' | grep 'Dypa' | grep -v 'Froc' | grep 'Scch' | grep 'Haha' | grep 'Maqu' | grep 'Mype' | grep 'Nete' | grep 'Nevi' | grep 'Nilu' | grep 'Pave' | grep 'Plci' | grep 'Ripe' | grep 'Siav' | grep 'Sifl' | grep 'Trur' | tr '\t' '\n' | grep -E "^$species" | tr ',' '\n' | sort | uniq | wc -l)
echo $species : $x
done

#heteroptera
for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
x=$(awk '{count = 0;if ($0 ~ /Ripe/) count++;if ($0 ~ /Nevi/) count++;if ($0 ~ /Cile/) count++;if ($0 ~ /Nete/) count++;if ($0 ~ /Aplu/) count++;if ($0 ~ /Haha/) count++;if (count >= 5) print $0;}' $orthofinder_results | tr '\t' '\n' | grep -E "^$species" | tr ',' '\n' | sort | uniq | wc -l)
echo $species : $x
done

#pyllid
for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
x=$(awk '{count = 0;if ($0 ~ /Pave/) count++;if ($0 ~ /Dici/) count++;if ($0 ~ /Trur/) count++;if ($0 ~ /Baco/) count++;if ($0 ~ /Dyap/) count++;if ($0 ~ /Dypa/) count++;if (count >= 5) print $0;}' $orthofinder_results | tr '\t' '\n' | grep -E "^$species" | tr ',' '\n' | sort | uniq | wc -l)
echo $species : $x
done

#aphid
for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
x=$(awk '{count = 0;if ($0 ~ /Scch/) count++;if ($0 ~ /Sifl/) count++;if ($0 ~ /Apgl/) count++;if ($0 ~ /Mype/) count++;if ($0 ~ /Acpi/) count++;if ($0 ~ /Siav/) count++;if (count >= 5) print $0;}' $orthofinder_results | tr '\t' '\n' | grep -E "^$species" | tr ',' '\n' | sort | uniq | wc -l)
echo $species : $x
done

#heteroptera exclusive
for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
x=$(awk '{count = 0;if ($0 ~ /Ripe/) count++;if ($0 ~ /Nevi/) count++;if ($0 ~ /Cile/) count++;if ($0 ~ /Nete/) count++;if ($0 ~ /Aplu/) count++;if ($0 ~ /Haha/) count++;if (count >= 5) print $0;}' $orthofinder_results | grep -v 'Acpi' | grep -v 'Apgl' | grep -v 'Baco' | grep -v 'Beta' | grep -v 'Dici' | grep -v 'Dyap' | grep -v 'Dypa' | grep -v 'Froc' | grep -v 'Scch' | grep -v 'Maqu' | grep -v 'Mype' | grep -v 'Nilu' | grep -v 'Pave' | grep -v 'Plci' | grep -v 'Siav' | grep -v 'Sifl' | grep -v 'Trur' | tr '\t' '\n' | grep -E "^$species" | tr ',' '\n' | sort | uniq | wc -l)
echo $species : $x
done

#psyllid exclusive
for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
x=$(awk '{count = 0;if ($0 ~ /Pave/) count++;if ($0 ~ /Dici/) count++;if ($0 ~ /Trur/) count++;if ($0 ~ /Baco/) count++;if ($0 ~ /Dyap/) count++;if ($0 ~ /Dypa/) count++;if (count >= 5) print $0;}' $orthofinder_results | grep -v 'Acpi' | grep -v 'Apgl' | grep -v 'Aplu' | grep -v 'Beta' | grep -v 'Cile' | grep -v 'Froc' | grep -v 'Haha' | grep -v 'Scch' | grep -v 'Maqu' | grep -v 'Mype' | grep -v 'Nete' | grep -v 'Nevi' | grep -v 'Nilu' | grep -v 'Plci' | grep -v 'Ripe' | grep -v 'Siav' | grep -v 'Sifl' | tr '\t' '\n' | grep -E "^$species" | tr ',' '\n' | sort | uniq | wc -l)
echo $species : $x
done

#aphid exclusive
for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
x=$(awk '{count = 0;if ($0 ~ /Scch/) count++;if ($0 ~ /Sifl/) count++;if ($0 ~ /Apgl/) count++;if ($0 ~ /Mype/) count++;if ($0 ~ /Acpi/) count++;if ($0 ~ /Siav/) count++;if (count >= 5) print $0;}' $orthofinder_results | grep -v 'Aplu' | grep -v 'Baco' | grep -v 'Beta' | grep -v 'Cile' | grep -v 'Dici' | grep -v 'Dyap' | grep -v 'Dypa' | grep -v 'Froc' | grep -v 'Haha' | grep -v 'Maqu' | grep -v 'Nete' | grep -v 'Nevi' | grep -v 'Nilu' | grep -v 'Pave' | grep -v 'Plci' | grep -v 'Ripe' | grep -v 'Trur' | tr '\t' '\n' | grep -E "^$species" | tr ',' '\n' | sort | uniq | wc -l)
echo $species : $x
done

#lineage specific
all_species="Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur"
for species in $all_species; do
    exclude_pattern=$(echo $all_species| sed 's/ /|/g' | sed "s/\b$species|\b//g" )
    x=$(grep -E "$species" "$orthofinder_results" | grep -vE "$exclude_pattern"| tr ',' '\n' | sort | uniq | wc -l)
    echo $species : $x
done

#Singletons:
head -n 11 /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23_hemiptera-3/formatted/orthofinder52000/Results_23_hemiptera/Comparative_Genomics_Statistics/Statistics_PerSpecies.tsv

#phloem feeder
for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
x=$(awk '{count = 0;if ($0 ~ /Pave/) count++;if ($0 ~ /Dici/) count++;if ($0 ~ /Trur/) count++;if ($0 ~ /Baco/) count++;if ($0 ~ /Dyap/) count++;if ($0 ~ /Dypa/) count++;if (count >= 5) print $0;}' $orthofinder_results | awk '{count = 0;if ($0 ~ /Scch/) count++;if ($0 ~ /Sifl/) count++;if ($0 ~ /Apgl/) count++;if ($0 ~ /Mype/) count++;if ($0 ~ /Acpi/) count++;if ($0 ~ /Siav/) count++;if (count >= 5) print $0;}' | grep -v 'Cile' | grep -v 'Aplu' | grep -v 'Haha' | grep -v 'Nete' | grep -v 'Nevi' | grep -v 'Ripe' | grep -v 'Froc' | tr '\t' '\n' | grep -E "^$species" | tr ',' '\n' | sort | uniq | wc -l)
echo $species : $x
done

Acpi : 98 #
Apgl : 74 #
Aplu : 0
Baco : 106 #
Beta : 41
Cile : 0
Dici : 81 #
Dyap : 72 #
Dypa : 66 #
Froc : 0
Haha : 0
Scch : 70 #
Maqu : 60
Mype : 109 #
Nete : 0
Nevi : 0
Nilu : 69 
Pave : 89 #
Plci : 56
Ripe : 0
Siav : 68 #
Sifl : 86 #
Trur : 63 #
#mean=81.8

for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
awk '{count = 0;if ($0 ~ /Pave/) count++;if ($0 ~ /Dici/) count++;if ($0 ~ /Trur/) count++;if ($0 ~ /Baco/) count++;if ($0 ~ /Dyap/) count++;if ($0 ~ /Dypa/) count++;if (count >= 5) print $0;}' $orthofinder_results | awk '{count = 0;if ($0 ~ /Scch/) count++;if ($0 ~ /Sifl/) count++;if ($0 ~ /Apgl/) count++;if ($0 ~ /Mype/) count++;if ($0 ~ /Acpi/) count++;if ($0 ~ /Siav/) count++;if (count >= 5) print $0;}' | grep -v 'Cile' | grep -v 'Aplu' | grep -v 'Haha' | grep -v 'Nete' | grep -v 'Nevi' | grep -v 'Ripe' | grep -v 'Froc' | tr '\t' '\n' | grep -E "OG" >> temp.txt
done
cat temp.txt | sort | uniq | wc -l #45
rm temp.txt

#not aphid
for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
x=$(awk '{count = 0;if ($0 ~ /Pave/) count++;if ($0 ~ /Dici/) count++;if ($0 ~ /Trur/) count++;if ($0 ~ /Baco/) count++;if ($0 ~ /Dyap/) count++;if ($0 ~ /Dypa/) count++;if (count >= 5) print $0;}' $orthofinder_results | awk '{count = 0;if ($0 ~ /Cile/) count++;if ($0 ~ /Aplu/) count++;if ($0 ~ /Haha/) count++;if ($0 ~ /Nete/) count++;if ($0 ~ /Nevi/) count++;if ($0 ~ /Ripe/) count++;if (count >= 5) print $0;}' | grep -v 'Scch' | grep -v 'Sifl' | grep -v 'Apgl' | grep -v 'Mype' | grep -v 'Acpi' | grep -v 'Siav' | grep -v 'Froc' | tr '\t' '\n' | grep -E "^$species" | tr ',' '\n' | sort | uniq | wc -l)
echo $species : $x
done

Acpi : 0
Apgl : 0
Aplu : 96 #
Baco : 232 #
Beta : 53
Cile : 113 #
Dici : 95 #
Dyap : 81 #
Dypa : 125 #
Froc : 0
Haha : 125 #
Scch : 0
Maqu : 157
Mype : 0
Nete : 81 #
Nevi : 86 #
Nilu : 112
Pave : 75 #
Plci : 33
Ripe : 89 #
Siav : 0
Sifl : 0
Trur : 66 #
#mean=94.9

for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
awk '{count = 0;if ($0 ~ /Pave/) count++;if ($0 ~ /Dici/) count++;if ($0 ~ /Trur/) count++;if ($0 ~ /Baco/) count++;if ($0 ~ /Dyap/) count++;if ($0 ~ /Dypa/) count++;if (count >= 5) print $0;}' $orthofinder_results | awk '{count = 0;if ($0 ~ /Cile/) count++;if ($0 ~ /Aplu/) count++;if ($0 ~ /Haha/) count++;if ($0 ~ /Nete/) count++;if ($0 ~ /Nevi/) count++;if ($0 ~ /Ripe/) count++;if (count >= 5) print $0;}' | grep -v 'Scch' | grep -v 'Sifl' | grep -v 'Apgl' | grep -v 'Mype' | grep -v 'Acpi' | grep -v 'Siav' | grep -v 'Froc' | tr '\t' '\n' | grep -E "OG" >> temp.txt
done
cat temp.txt | sort | uniq | wc -l #50
rm temp.txt

#not psyllid
for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
x=$(awk '{count = 0;if ($0 ~ /Cile/) count++;if ($0 ~ /Aplu/) count++;if ($0 ~ /Haha/) count++;if ($0 ~ /Nete/) count++;if ($0 ~ /Nevi/) count++;if ($0 ~ /Ripe/) count++;if (count >= 5) print $0;}' $orthofinder_results | awk '{count = 0;if ($0 ~ /Scch/) count++;if ($0 ~ /Sifl/) count++;if ($0 ~ /Apgl/) count++;if ($0 ~ /Mype/) count++;if ($0 ~ /Acpi/) count++;if ($0 ~ /Siav/) count++;if (count >= 5) print $0;}' | grep -v 'Pave' | grep -v 'Dici' | grep -v 'Trur' | grep -v 'Baco' | grep -v 'Dyap' | grep -v 'Dypa' | grep -v 'Froc' | tr '\t' '\n' | grep -E "^$species" | tr ',' '\n' | sort | uniq | wc -l)
echo $species : $x
done

Acpi : 55 #
Apgl : 29 #
Aplu : 38 #
Baco : 0
Beta : 25
Cile : 56 #
Dici : 0
Dyap : 0
Dypa : 0
Froc : 0
Haha : 70 #
Scch : 31 #
Maqu : 56
Mype : 37 #
Nete : 29 #
Nevi : 40 #
Nilu : 47
Pave : 0
Plci : 29
Ripe : 33 #
Siav : 39 #
Sifl : 49 #
Trur : 0
#mean=42.2

for species in Acpi Apgl Aplu Baco Beta Cile Dici Dyap Dypa Froc Haha Scch Maqu Mype Nete Nevi Nilu Pave Plci Ripe Siav Sifl Trur; do
awk '{count = 0;if ($0 ~ /Cile/) count++;if ($0 ~ /Aplu/) count++;if ($0 ~ /Haha/) count++;if ($0 ~ /Nete/) count++;if ($0 ~ /Nevi/) count++;if ($0 ~ /Ripe/) count++;if (count >= 5) print $0;}' $orthofinder_results | awk '{count = 0;if ($0 ~ /Scch/) count++;if ($0 ~ /Sifl/) count++;if ($0 ~ /Apgl/) count++;if ($0 ~ /Mype/) count++;if ($0 ~ /Acpi/) count++;if ($0 ~ /Siav/) count++;if (count >= 5) print $0;}' | grep -v 'Pave' | grep -v 'Dici' | grep -v 'Trur' | grep -v 'Baco' | grep -v 'Dyap' | grep -v 'Dypa' | grep -v 'Froc' | tr '\t' '\n' | grep -E "OG" >> temp.txt
done
cat temp.txt | sort | uniq | wc -l #30
rm temp.txt

#R.pedestris - plant cell rupture/xylem ingestion 10.3390/insects13060511 phloem
#H. halys - phloem?
#N. viridula - cell rupture 10.3954/JAUE18-09.1
#C.lectularius - blood feeder
#A. lucorum - plant cell rupture 10.1093/jisesa/ieab007 Feeding Behavior of Riptortus pedestris (Fabricius) on Soybean: Electrical Penetration Graph Analysis and Histological Investigations
#N.tenuis - predatory bug + cell rupture
```
```bash
#Get orthogroups for D.apicales, D.pallida, and Dyspera clades that are significantly (p<=0.01) expanded or contracted:
cafe_results=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/22_hemiptera+Froc_results/result_summary.tsv
orthofinder_results=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthofinder/23_hemiptera-3/formatted/orthofinder52000/Results_23_hemiptera/Orthogroups/Orthogroups.tsv
OutDir=/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/22_hemiptera+Froc_results

#Contracted - get Froc genes for orthogroups reduced
awk -F'\t' '$2 == "39" && $5 < 0.01 && $4 < 0 {print $1}' $cafe_results | xargs -I {} grep {} $orthofinder_results | tr '\t' '\n' | grep '^Froc' | tr ',' '\n'  | cut -d '|' -f2 > ${OutDir}/contracted_Dy_node_Froc_genes.txt
awk -F'\t' '$2 == "Dyap" && $5 < 0.01 && $4 < 0 {print $1}' $cafe_results | xargs -I {} grep {} $orthofinder_results | tr '\t' '\n' | grep '^Froc' | tr ',' '\n'  | cut -d '|' -f2 > ${OutDir}/contracted_Dyap_node_Froc_genes.txt
awk -F'\t' '$2 == "Dypa" && $5 < 0.01 && $4 < 0 {print $1}' $cafe_results | xargs -I {} grep {} $orthofinder_results | tr '\t' '\n' | grep '^Froc' | tr ',' '\n'  | cut -d '|' -f2 > ${OutDir}/contracted_Dypa_node_Froc_genes.txt

#Expanded
awk -F'\t' '$2 == "39" && $5 < 0.01 && $4 > 0 {print $1}' $cafe_results | xargs -I {} grep {} $orthofinder_results | tr '\t' '\n' | grep '^Dypa' | tr ',' '\n'  | cut -d '|' -f2 | awk -F'\t' -v OFS='\t' '{ $1 = $1 "_dypa"; print }' > ${OutDir}/expanded_Dy_node_genes.txt
awk -F'\t' '$2 == "39" && $5 < 0.01 && $4 > 0 {print $1}' $cafe_results | xargs -I {} grep {} $orthofinder_results | tr '\t' '\n' | grep '^Dyap' | tr ',' '\n'  | cut -d '|' -f2 | awk -F'\t' -v OFS='\t' '{ $1 = $1 "_dyap"; print }' >> ${OutDir}/expanded_Dy_node_genes.txt

awk -F'\t' '$2 == "Dyap" && $5 < 0.01 && $4 > 0 {print $1}' $cafe_results | xargs -I {} grep {} $orthofinder_results | tr '\t' '\n' | grep '^Dyap' | tr ',' '\n'  | cut -d '|' -f2 > ${OutDir}/expanded_Dyap_node_genes.txt
awk -F'\t' '$2 == "Dypa" && $5 < 0.01 && $4 > 0 {print $1}' $cafe_results | xargs -I {} grep {} $orthofinder_results | tr '\t' '\n' | grep '^Dypa' | tr ',' '\n'  | cut -d '|' -f2 > ${OutDir}/expanded_Dypa_node_genes.txt

awk -F'\t' -v OFS='\t' '{ if ($1 ~ /^g/) $1 = $1 "_dyap"; print }' analysis/eggnog/Dyspera_apicales_eggnog_mapper_files/Dyspera_apicales_out.emapper.annotations.txt > /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/22_hemiptera+Froc_results/Dyspera_out.emapper.annotations2.txt
awk -F'\t' -v OFS='\t' '{ if ($1 ~ /^g/) $1 = $1 "_dypa"; print }' analysis/eggnog/Dyspera_pallida_eggnog_mapper_files/Dyspera_pallida_out.emapper.annotations.txt | grep '_dypa' >> /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/22_hemiptera+Froc_results/Dyspera_out.emapper.annotations2.txt #convert to xlsx
```
```bash
download_dir="Sifl_GCF_003268045.1_eggnog_mapper_files"
mkdir -p "$download_dir"
cd "$download_dir" || exit
wget -r -np -nH --cut-dirs=1 -R "index.html*" http://eggnog-mapper.embl.de/MM_rzxp233j/
for file in *; do
  mv "$file" "Sifl_GCF_003268045.1_$file"
done
echo "Download and renaming complete."
cd ..
```
```bash
ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/eggnog2/Dyap_Dyspera_apicales_eggnog_mapper_files/Dyap_Dyspera_apicales_out.emapper.annotations.xlsx
ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/eggnog2/Dypa_Dyspera_pallida_eggnog_mapper_files/Dypa_Dyspera_pallida_out.emapper.annotations.xlsx
ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/eggnog2/Froc_GCF_000697945.3_eggnog_mapper_files/Froc_GCF_000697945.3_out.emapper.annotations.xlsx
```
```R
if (!requireNamespace("BiocManager", quietly = TRUE)) 
         install.packages("BiocManager")

BiocManager::install("clusterProfiler")
BiocManager::install("enrichplot")
BiocManager::install("topGO")
install.packages("data.table")
install.packages("readxl")

library(clusterProfiler)
library(enrichplot)
library(data.table)
library(readxl)
library(stringr)
library(tidyverse)
library(topGO)
library(lintr)
library(lattice)
library(dplyr)

#Set data
#annotation <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/eggnog2/Dyap_Dyspera_apicales_eggnog_mapper_files/Dyap_Dyspera_apicales_out.emapper.annotations.xlsx"
#protein <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/22_hemiptera+Froc_results/expanded_Dyap_node_genes.txt"
annotation <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/eggnog2/Dypa_Dyspera_pallida_eggnog_mapper_files/Dypa_Dyspera_pallida_out.emapper.annotations.xlsx"
protein <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/22_hemiptera+Froc_results/expanded_Dypa_node_genes.txt"
#annotation <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/22_hemiptera+Froc_results/Dyspera_out.emapper.annotations2.xlsx"
#protein <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/22_hemiptera+Froc_results/expanded_Dy_node_genes.txt"
#annotation <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/eggnog2/Froc_GCF_000697945.3_eggnog_mapper_files/Froc_GCF_000697945.3_out.emapper.annotations.xlsx"
#protein <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/22_hemiptera+Froc_results/contracted_Dy_node_Froc_genes.txt"
#annotation <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/eggnog2/Froc_GCF_000697945.3_eggnog_mapper_files/Froc_GCF_000697945.3_out.emapper.annotations.xlsx"
#protein <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/22_hemiptera+Froc_results/contracted_Dyap_node_Froc_genes.txt"
#annotation <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/eggnog2/Froc_GCF_000697945.3_eggnog_mapper_files/Froc_GCF_000697945.3_out.emapper.annotations.xlsx"
#protein <- "//jic-hpc-data/Group-Scratch/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/cafe/22_hemiptera+Froc_results/contracted_Dypa_node_Froc_genes.txt"

#Read in the data
ko_output_file_path <- gsub("genes.txt", "_KOEnr_results.csv", protein)
ko_plot_output_file_path <- gsub("genes.txt", "_KOEnr_plot.tiff", protein)
go_output_file_path <- gsub("genes.txt", "_GOEnr_results.csv", protein)
go_plot_output_file_path <- gsub("genes.txt", "_GOEnr_plot.tiff", protein)

eggnog_data <- read_excel(annotation, skip = 2)
eggnog_data$query <- sapply(strsplit(eggnog_data$query, "\\|"), `[`, 1)
proteins<- read.table(protein, 
                       header = FALSE, 
                       sep = "\t",
                       stringsAsFactors = FALSE)
protein_ids <- proteins[[1]]

#get columns 1 (query) and 12 (KO terms) for data rows
kegg_data <- eggnog_data[c(1,12)]
go_data <- eggnog_data[c(1,10)]

#clean up by removing the "ko:" in front of every KO term
kegg_data$KEGG_ko <- gsub("ko:", "", kegg_data$KEGG_ko)

#expand, since some genes/proteins will have multiple assigned KO terms
kegg_data <- data.table(kegg_data)
kegg_data <- kegg_data[, list(KEGG_ko = unlist(strsplit(KEGG_ko , ","))), by = query]
go_data <- data.table(go_data)
go_data <- go_data[, list(GOs = unlist(strsplit(GOs , ","))), by = query]

#select the needed columns
kegg_final <- kegg_data[,c(2,1)]
go_final <- go_data[,c(2,1)]

#Enrichment analyses using a mininum size of genes/proteins annotated for testing of 10 (parameter minGSSize), a p-value cut-off of 0.05 and adjust p-values using the Benjamini-Hochberg method with a q-value cut-off of 0.05.
ko_enr_res <- enricher(protein_ids, TERM2GENE=kegg_final, pvalueCutoff = 0.05, pAdjustMethod = "BH", qvalueCutoff = 0.05, minGSSize = 10)
go_enr_res <- enricher(protein_ids, TERM2GENE=go_final, pvalueCutoff = 0.05, pAdjustMethod = "BH", qvalueCutoff = 0.05, minGSSize = 10)

#output the results of the analysis
write.table(ko_enr_res, file = ko_output_file_path, sep = ",", row.names = FALSE, quote = FALSE)
write.table(go_enr_res, file = go_output_file_path, sep = ",", row.names = FALSE, quote = FALSE)

# make a simple dot plot of the results and save it as a .tiff file
tiff(file=ko_plot_output_file_path,width=6, height=4, units="in", res=1200)
dotplot(ko_enr_res, showCategory=30)
dev.off()
tiff(file=go_plot_output_file_path,width=6, height=4, units="in", res=1200)
dotplot(go_enr_res, showCategory=30)
dev.off()
```
#### Orthovenn

```bash
singularity pull docker://lufang0411/orthovenn3-mysql:latest
singularity pull docker://lufang0411/orthovenn3-api:latest
singularity pull docker://leeoluo/orthovenn3-front:latest
mkdir -p ~/data/venn3/api
mkdir -p ~/data/venn3/mysql
singularity run --bind ~/data/venn3/mysql:/var/lib/mysql \
--env MYSQL_ROOT_PASSWORD=123456Aa \
orthovenn3-mysql_latest.sif

singularity run --bind ~/data/venn3/api/result:/home/orthovenn3/result --bind ~/data/venn3/api/work:/home/orthovenn3/work orthovenn3-api_latest.sif --restart=always -p 8102:8102 -p 8122:22 --name orthovenn3-api --link orthovenn3-mysql





mkdir /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3

sudo docker pull lufang0411/orthovenn3-mysql:latest &&
sudo docker pull lufang0411/orthovenn3-api:latest &&
sudo docker pull leeoluo/orthovenn3-front:latest

mkdir -p ~/data/venn3/api && mkdir -p ~/data/venn3/mysql

sudo docker run --privileged \
--restart=always \
--name orthovenn3-mysql \
-e MYSQL_ROOT_PASSWORD=123456Aa \
-v ~/data/venn3/mysql:/var/lib/mysql \
-d lufang0411/orthovenn3-mysql

sudo docker run -dit  \
--restart=always \
-p 8102:8102 \
-p 8122:22 \
--name orthovenn3-api \
--link orthovenn3-mysql \
-v ~/data/venn3/api/result:/home/orthovenn3/result \
-v ~/data/venn3/api/work:/home/orthovenn3/work \
lufang0411/orthovenn3-api /docker-entrypoint.sh

ifconfig

sudo docker run -d \
--restart=always \
--name orthovenn3-web7 \
-p 9990:80 \
--link orthovenn3-api \
-e API_URL=http://172.30.242.48:8102 \
-v ~/data/venn3/api/result:/data/orthovenn3/api/result \
leeoluo/orthovenn3-front:latest

http://172.30.242.48:9990


/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/braker.aa
/jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/braker.aa

cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_anthrisci/hifiasm_19.5/820m/48/1/10.0/0.25/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/braker.aa /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Dyspera_pallida.faa
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_apicales/hifiasm_19.5/880m/29/3/3.0/0.75/break10x/purge_dups/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+tsebra_loose/braker.aa /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Dyspera_apicales.faa
cp /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/assembly/genome/T_urticae/hifiasm_19.5/715m/12/2/3.0/0.5/break10x/purge_dups/purge_haplotigs/yahs/sanger/MitoHifi/filtered/inspector/repeatmasker/softmask/braker3+helixer+loose/braker.aa /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/Trioza_urticae.faa

for file in $(ls /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/Psyllidae/analysis/orthovenn3/*.faa); do
OutDir=$(dirname $file)
OutFile=$(basename $file | sed 's@.faa@_longest.faa@g')
singularity exec /jic/scratch/groups/Saskia-Hogenhout/tom_heaven/containers/python3.sif python3 ~/git_repos/Scripts/NBI/longest_variant.py $file ${OutDir}/${OutFile}
done

sudo docker run --name venns-front -p 9989:80 --link venns -v /your_work_path:/data/orthovenn2 -e API_HOST=venns -e API_PORT=6001 -d lufang0411/orthovenn2-front:latest

http://localhost:9989



sudo docker container stop orthovenn3-web8
sudo docker container rm orthovenn3-web8

sudo docker image rm leeoluo/orthovenn3-front:latest
sudo docker pull leeoluo/orthovenn3-front:latest

sudo docker run -d \
--restart=always \
--name orthovenn3-web \
-p 9998:80 \
--link orthovenn3-api \
-e API_URL=http://172.30.242.48:8102 \
-v ~/data/venn3/api/result:/data/orthovenn3/api/result \
leeoluo/orthovenn3-front:latest


######################################################################################################################
mkdir -p /home/theaven/scratch/apps/containers/mongo
mkdir -p /home/theaven/scratch/apps/containers/orthovenn2

# Pull images
singularity pull docker://mongo:latest
singularity pull docker://lufang0411/orthovenn2:latest
singularity pull docker://lufang0411/orthovenn2-front:latest

# Start MongoDB instance
singularity instance start --writable-tmpfs --bind /home/theaven/scratch/apps/containers/mongo:/data/db mongo_latest.sif venns-mongo
singularity exec instance://venns-mongo /usr/bin/mongod --dbpath /data/db --port 27017 &




# Start Orthovenn2 API instance
singularity instance start --writable-tmpfs --bind /your_work_path:/data/orthovenn2 orthovenn2_latest.sif venns
singularity exec instance://venns /path/to/entrypoint --port 6001 --mongo_host 127.0.0.1 &

# Start Orthovenn2 Front instance
singularity instance start --writable-tmpfs --bind /your_work_path:/data/orthovenn2 orthovenn2_front_latest.sif venns-front
singularity exec instance://venns-front /path/to/entrypoint --api_host 127.0.0.1 --api_port 6001 &


sudo docker run --name venns-front -p 9899:80 --link venns -v ~/data/venn2:/data/orthovenn2 -e API_HOST=venns -e API_PORT=6001 -d lufang0411/orthovenn2-front:latest
```
Orthovenn3
```bash
ssh -i "testkeypair.pem" ec2-user@ec2-54-84-46-131.compute-1.amazonaws.com
sudo yum install docker
sudo systemctl start docker
sudo systemctl enable docker
sudo systemctl status docker
sudo reboot
ssh -i "testkeypair.pem" ec2-user@ec2-54-84-46-131.compute-1.amazonaws.com
sudo docker pull lufang0411/orthovenn3-mysql:latest &&
sudo docker pull lufang0411/orthovenn3-api:latest &&
sudo docker pull leeoluo/orthovenn3-front:latest
mkdir -p ~/data/venn3/api && mkdir -p ~/data/venn3/mysql
sudo docker run --privileged \
--restart=always \
--name orthovenn3-mysql \
-e MYSQL_ROOT_PASSWORD=123456Aa \
-v ~/data/venn3/mysql:/var/lib/mysql \
-d lufang0411/orthovenn3-mysql
sudo docker run -dit  \
--restart=always \
-p 8102:8102 \
-p 8122:22 \
--name orthovenn3-api \
--link orthovenn3-mysql \
-v ~/data/venn3/api/result:/home/orthovenn3/result \
-v ~/data/venn3/api/work:/home/orthovenn3/work \
lufang0411/orthovenn3-api /docker-entrypoint.sh

sudo docker run -d \
--restart=always \
--name orthovenn3-web \
-p 9994:80 \
--link orthovenn3-api \
-e API_URL=http://54.235.1.132:8102 \
-v ~/data/venn3/api/result:/data/orthovenn3/api/result \
leeoluo/orthovenn3-front:latest

http://54.235.1.132:9994

df -h
top
free -h
```
Orthovenn2
```bash
ssh -i "testkeypair.pem" ec2-user@ec2-54-235-1-132.compute-1.amazonaws.com
sudo yum install docker
sudo systemctl start docker
sudo systemctl enable docker
sudo systemctl status docker
sudo reboot
ssh -i "testkeypair.pem" ec2-user@ec2-54-235-1-132.compute-1.amazonaws.com
sudo docker run --name venns-mongo -v /your_mongodb_path:/data/db -d -p 27017:27017 mongo:latest
sudo docker run -d --name venns -p 6001:6001 --link venns-mongo -v ~/data/venn2:/data/orthovenn2 -e MONGO_HOST=venns-mongo lufang0411/orthovenn2:latest
sudo docker run --name venns-front -p 9995:80 --link venns -v ~/data/venn2:/data/orthovenn2 -e API_HOST=venns -e API_PORT=6001 -d lufang0411/orthovenn2-front:latest
http://54.235.1.132:9995
```